<!doctype html>

<html lang="en-us">

<head>
  <title>BMMB 554 | 2018</title>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="description" content="The HTML5 Herald" />
  <meta name="author" content="Anton Nekrutenko" />
  <meta name="generator" content="Hugo 0.32.3" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/normalize/7.0.0/normalize.min.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" />
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto+Slab|Ruda" />
  <link rel="stylesheet" type="text/css" href="https://nekrut.github.io/BMMB554/css/styles.css" />
</head>

<body>
  <div id="container">
    <header>
      <h1>
                <a href="https://nekrut.github.io/BMMB554/">BMMB 554 | 2018</a>
            </h1>

      <ul id="social-media">
        
        <li><a href="https://twitter.com/galaxyproject"><i class="fa fa-twitter fa-lg" aria-hidden="true"></i></a></li>
          
        <li><a href="https://github.com/nekrut"><i class="fa fa-github fa-lg" aria-hidden="true"></i></a></li>
           
      </ul>
      
      <p><em>Introduction to data driven life sciences</em></p>
      
    </header>

    
<nav>
    <ul>
        
        <li>
            <a class="" href="https://nekrut.github.io/BMMB554/about/">
                <i class="fa-li fa  fa-lg"></i><span>Syllabus</span>
            </a>
        </li>
        
        <li>
            <a class="" href="https://nekrut.github.io/BMMB554/posts/">
                <i class="fa-li fa  fa-lg"></i><span>Lectures</span>
            </a>
        </li>
        
    </ul>
</nav>

    <main>




<article>

    <h1>Continuing MEGA analysis</h1>

    
        <aside>
    <ul>
        <li>
            <time class="post-date" datetime="2018-03-28T00:00:00Z">Mar 28, 2018</time>
        </li>
        
        

        
        <li>
            <em>
                
                    
                    <a href="https://nekrut.github.io/BMMB554/tags/writing/">#writing</a>
                
                    , 
                    <a href="https://nekrut.github.io/BMMB554/tags/papers/">#papers</a>
                
                    , 
                    <a href="https://nekrut.github.io/BMMB554/tags/grants/">#grants</a>
                
            </em>
        </li>
        

        <li>1 min read</li>
    </ul>
</aside>
    

    

<h1 id="refresher">Refresher</h1>

<p>We were analyzing the following data:</p>

<table>
<thead>
<tr>
<th></th>
</tr>
</thead>

<tbody>
<tr>
<td><img src="https://nekrut.github.io/BMMB554/img/team_data.png" alt="" /></td>
</tr>

<tr>
<td><small>Trajectories for the four teams in this class:</small></td>
</tr>
</tbody>
</table>

<table>
<thead>
<tr>
<th>Team 1</th>
<th>Team 2</th>
<th>Team 3</th>
<th>Team 4</th>
</tr>
</thead>

<tbody>
<tr>
<td>10.SRR3722230</td>
<td>103.SRR3721968</td>
<td>100.SRR3722240</td>
<td>2.SRR3721964</td>
</tr>

<tr>
<td>2.SRR3721964</td>
<td>104.SRR3721969</td>
<td>101.SRR3721966</td>
<td>34.SRR3722148</td>
</tr>

<tr>
<td>36.SRR3722150</td>
<td>105.SRR3721970</td>
<td>122.SRR3721989</td>
<td>38.SRR3722152</td>
</tr>

<tr>
<td>37.SRR3722151</td>
<td>106.SRR3721971</td>
<td>2.SRR3721964</td>
<td>43.SRR3722158</td>
</tr>

<tr>
<td>64.SRR3722201</td>
<td>168.SRR3722039</td>
<td>74.SRR3722212</td>
<td>63.SRR3722200</td>
</tr>

<tr>
<td>7.SRR3722197</td>
<td>169.SRR3722040</td>
<td>95.SRR3722235</td>
<td>67.SRR3722204</td>
</tr>

<tr>
<td>8.SRR3722208</td>
<td></td>
<td>96.SRR3722236</td>
<td>68.SRR3722205</td>
</tr>

<tr>
<td>9.SRR3722219</td>
<td></td>
<td>97.SRR3722237</td>
<td>69.SRR3722206</td>
</tr>

<tr>
<td></td>
<td></td>
<td>98.SRR3722238</td>
<td>7.SRR3722197</td>
</tr>

<tr>
<td></td>
<td></td>
<td>99.SRR3722239</td>
<td>70.SRR3722207</td>
</tr>

<tr>
<td></td>
<td></td>
<td></td>
<td>71.SRR3722209</td>
</tr>

<tr>
<td></td>
<td></td>
<td></td>
<td>72.SRR3722210</td>
</tr>

<tr>
<td></td>
<td></td>
<td></td>
<td>73.SRR3722211</td>
</tr>
</tbody>
</table>

<p>The data is at <a href="https://usegalaxy.org/library/list#folders/F9b810fe4cca869a8">this location</a>:</p>

<table>
<thead>
<tr>
<th></th>
</tr>
</thead>

<tbody>
<tr>
<td><img src="https://nekrut.github.io/BMMB554/img/mega_data.png" alt="" /></td>
</tr>

<tr>
<td><small>Folders T1 through T4 contain data for each team. <code>U00096</code> is a Genbank file containing annotations and genome sequence for <em>E. coli</em> genome used by <a href="http://science.sciencemag.org/content/353/6304/1147">Baym:2016</a> (see Methods section).</small></td>
</tr>
</tbody>
</table>

<h1 id="analysis">Analysis</h1>

<h2 id="initial-processing-and-finding-variants">Initial processing and finding variants</h2>

<p>For initial analysis (mapping and variant calling) we used the following <a href="https://usegalaxy.org/u/aun1/w/megatovcf">workflow</a>:</p>

<table>
<thead>
<tr>
<th></th>
</tr>
</thead>

<tbody>
<tr>
<td><img src="https://nekrut.github.io/BMMB554/img/mega_workflow.png" alt="" /></td>
</tr>

<tr>
<td><small>Workflow for the initial MEGA analysis</small></td>
</tr>
</tbody>
</table>

<p>This workflow contains the following steps:</p>

<ol>
<li>Inputs:

<ul>
<li>A collection of fastq datasets</li>
<li>A reference genome in Genbank format</li>
</ul></li>
<li>Genbank dataset in processed by <strong>SnpEff Build</strong> tool to generate two outputs:

<ul>
<li>FASTA dataset</li>
<li>snpEff annotation database</li>
</ul></li>
<li>Mapping against <em>E. coli</em> with BWA-MEM</li>
<li>Deduplcating reads</li>
<li>Filtering reads</li>
<li>Running <strong>Freebayes</strong></li>
</ol>

<p>Results of these analyses for individual teams are:</p>

<ul>
<li><a href="https://usegalaxy.org/u/aun1/h/t1-analysis">T1</a></li>
<li><a href="https://usegalaxy.org/u/aun1/h/t2-analysis">T2</a></li>
<li><a href="https://usegalaxy.org/u/aun1/h/t3-analysis">T3</a></li>
<li><a href="https://usegalaxy.org/u/aun1/h/t4-analysis">T4</a></li>
</ul>

<h2 id="postprocessing">Postprocessing</h2>

<p>For deciding which variants are &ldquo;good&rdquo; and understanding their effects we will use two toolkits we will try to do this:</p>

<ul>
<li>Run <a href="http://snpeff.sourceforge.net/SnpEff_manual.html">SnpEff</a></li>
<li>Run <a href="http://snpeff.sourceforge.net/SnpSift.html">SnpSift</a></li>
<li>Calculate <a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-13-666">Stand Bias</a></li>
<li>Analyze data in Jupyter</li>
</ul>


</article>


<section class="post-nav">
    <ul>
        
        <li>
            <a href="https://nekrut.github.io/BMMB554/posts/topic14/"><i class="fa fa-chevron-circle-left"></i> Finding DNA/Protein interactions</a>
        </li>
        
        
    </ul>
</section>
    





</main>
    <footer>
        <h6>Anton Nekrutenko | 2018 | CC-BY </h6>
    </footer>
</div>
<script src="https://nekrut.github.io/BMMB554/js/scripts.js"></script>
<script type="text/javascript" async
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
  });
  MathJax.Hub.Queue(function() {
    
    
    
    var all = MathJax.Hub.getAllJax(), i;
    for(i = 0; i < all.length; i += 1) {
        all[i].SourceElement().parentNode.className += ' has-jax';
    }
  });

  MathJax.Hub.Config({
  
  TeX: { equationNumbers: { autoNumber: "AMS" } }
  });
</script>

</body>
</html>


