<!DOCTYPE html>
<!--[if lt IE 7]><html class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
<!--[if IE 7]><html class="no-js lt-ie9 lt-ie8"> <![endif]-->
<!--[if IE 8]><html class="no-js lt-ie9"> <![endif]-->
<!--[if gt IE 8]><!--><html class="no-js" prefix="og: http://ogp.me/ns#" xmlns:og="http://ogp.me/ns#"><!--<![endif]-->

    <head>
                <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="description" content="**Topic 7** | Handling NGS data: from FASTQ to BAM">
        <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=1.0" />
        <meta name="mobile-web-app-capable" content="yes">
        <meta name="apple-mobile-web-app-capable" content="yes">
        <meta name="apple-mobile-web-app-status-bar-style" content="black">
        <meta name="author" content="Anton Nekrutenko">
  
	
        <meta property="og:site_name" content="BMMB 554 / 2017">
        <meta property="og:title" content="BMMB 554 / 2017">
        <meta property="og:url" content="https://nekrut.github.io/BMMB554/post/topic7/">
        <meta property="og:description" content="Introduction to data driven life sciences">
    
        <meta property="og:type" content="article" />
        <meta property="og:article:author" content="Anton Nekrutenko" />
        <meta property="og:article:published_time" content="2017-02-01T11:26:49-04:00" />
    
        <meta name="generator" content="Hugo 0.18.1" />
        <title>7. NGS data: Practicalities &middot; BMMB 554 / 2017</title>
        <link rel="canonical" href="https://nekrut.github.io/BMMB554/" />
        <link rel="alternate" type="application/rss+xml" title="RSS" href="">
        <link rel="stylesheet" type="text/css" href="https://nekrut.github.io/BMMB554/css/main.css"/>
        <link href="//fonts.googleapis.com/css?family=Source+Sans+Pro:200|Montserrat:700" rel="stylesheet" type="text/css">
        <link href="//netdna.bootstrapcdn.com/font-awesome/4.0.3/css/font-awesome.css" rel="stylesheet">
        <script src="//code.jquery.com/jquery-1.10.2.min.js"></script>
    </head>

<body>
<!--[if lt IE 7]><p class="browsehappy">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> or <a href="http://www.google.com/chrome/‎">install Google Chrome</a> to experience this site.</p><![endif]-->

    <header id="site-header">
        <div class="container">
            <a href="https://nekrut.github.io/BMMB554/" alt="BMMB 554 / 2017"><h1 class="blog-title heading">BMMB 554 / 2017</h1></a>
            <p class="blog-description">Introduction to data driven life sciences</p>
        </div>
    </header>
<main class="content" role="main">
	<div class="container">
		<article class="post">
	<header class="post-header">
        <h3 class="p-post-title">7. NGS data: Practicalities</h3>
    </header>

    <section class="post-content">
        

<p><img src="/BMMB554/img/topic7_cover.png" alt="" /></p>

<p>In this section we will look at practical aspects of manipulation of next-generation sequencing data. We will start with Fastq format produced by most sequencing machines and will finish with SAM/BAM format representing mapped reads. The cover image above shows a screen dump of a SAM dataset.</p>

<h1 id="getting-ngs-data-in">Getting NGS data in</h1>

<p>You can data in Galaxy using one of five ways:</p>

<h2 id="from-your-computer">From your computer</h2>

<p>This works well for small files because web browser do not like lengthy file transfers:</p>


<div style="position: relative; padding-bottom: 56.25%; padding-top: 30px; height: 0; overflow: hidden;">
  <iframe src="//player.vimeo.com/video/120901536" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
 </div>


<h2 id="using-ftp">Using FTP</h2>

<p>FTP (<a href="https://en.wikipedia.org/wiki/File_Transfer_Protocol">file transfer protocol</a>) allows transferring large collection of files:</p>


<div style="position: relative; padding-bottom: 56.25%; padding-top: 30px; height: 0; overflow: hidden;">
  <iframe src="//player.vimeo.com/video/120972739" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
 </div>


<h2 id="from-the-web">From the Web</h2>

<p>Upload from the web works when URL (addresses) of data files are known:</p>


<div style="position: relative; padding-bottom: 56.25%; padding-top: 30px; height: 0; overflow: hidden;">
  <iframe src="//player.vimeo.com/video/120973708" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
 </div>


<h2 id="from-ebi-short-read-archive">From EBI short read archive</h2>

<p>This is the best way to upload published datasets deposited to EBI SRA. The problem is that not all datasets are available from EBI. Next option (below) explain how to deal with NCBI SRA datasets:</p>


<div style="position: relative; padding-bottom: 56.25%; padding-top: 30px; height: 0; overflow: hidden;">
  <iframe src="//player.vimeo.com/video/121187220" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
 </div>


<h2 id="from-ncbi-short-read-archive">From NCBI short read archive</h2>

<p>Finally, datasets can be uploaded directly from NCBI&rsquo;s short read archive:</p>


<div style="position: relative; padding-bottom: 56.25%; padding-top: 30px; height: 0; overflow: hidden;">
  <iframe src="//player.vimeo.com/video/121190377" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
 </div>


<h3 id="font-color-red-9888-try-it-yourself-font"><font color="red">&#9888; Try it yourself</font></h3>

<ul>
<li>Create a new Galaxy history at <a href="http://usegalaxy.org">http://usegalaxy.org</a> (don&rsquo;t forget to log in).</li>
<li>Import the following two datasets (for help see the above video &ldquo;<em>From the Web</em>&rdquo;):</li>
</ul>

<blockquote>
<p><a href="http://www.bx.psu.edu/~anton/share/ng_test_data/var/raw_mother-ds-1.fq.gz">http://www.bx.psu.edu/~anton/share/ng_test_data/var/raw_mother-ds-1.fq.gz</a>
<a href="http://www.bx.psu.edu/~anton/share/ng_test_data/var/raw_mother-ds-2.fq.gz">http://www.bx.psu.edu/~anton/share/ng_test_data/var/raw_mother-ds-2.fq.gz</a></p>
</blockquote>

<p>These are paired end data (see below for explanation of what paired-end is) for a single Illumina run.</p>

<p>Keep Galaxy history for later. We will need it again in a few minutes.</p>

<h1 id="fastq-manipulation-and-quality-control">Fastq manipulation and quality control</h1>

<h2 id="what-is-fastq">What is Fastq?</h2>

<p><a href="http://en.wikipedia.org/wiki/FASTQ_format">FastQ</a> is not a very well defined format. In the beginning various manufacturers of sequencing instruments were free to interpret fastq as they saw fit, resulting in a multitude of fastq flavors. This variation stemmed primarily from different ways of encoding quality values as described <a href="http://en.wikipedia.org/wiki/FASTQ_format">here</a> (below you will explanation of quality scores and their meaning). Today, <a href="http://www.ncbi.nlm.nih.gov/pubmed/20015970">fastq Sanger</a> version of the format is considered to be the standard form of fastq. Galaxy is using fastq sanger as the only legitimate input for downstream processing tools and provides <a href="http://www.ncbi.nlm.nih.gov/pubmed/20562416">a number of utilities for converting fastq files</a> into this form (see <strong>NGS: QC and manipulation</strong> section of Galaxy tools).</p>

<p>Fastq format looks like this:</p>

<pre><code>
@M02286:19:000000000-AA549:1:1101:12677:1273 1:N:0:23
CCTACGGGTGGCAGCAGTGAGGAATATTGGTCAATGGACGGAAGTCTGAACCAGCCAAGTAGCGTGCAG
+
ABC8C,:@F:CE8,B-,C,-6-9-C,CE9-CC--C-&lt;-C++,,+;CE&lt;,,CD,CEFC,@E9&lt;FCFCF?9
@M02286:19:000000000-AA549:1:1101:15048:1299 1:N:0:23
CCTACGGGTGGCTGCAGTGAGGAATATTGGACAATGGTCGGAAGACTGATCCAGCCATGCCGCGTGCAG
+
ABC@CC77CFCEG;F9&lt;F89&lt;9--C,CE,--C-6C-,CE:++7:,CF&lt;,CEF,CFGGD8FFCFCFEGCF
@M02286:19:000000000-AA549:1:1101:11116:1322 1:N:0:23
CCTACGGGAGGCAGCAGTAGGGAATCTTCGGCAATGGACGGAAGTCTGACCGAGCAACGCCGCGTGAGT
+
AAC&lt;CCF+@@&gt;CC,C9,F9C9@9-CFFFE@7@:+CC8-C@:7,@EFE,6CF:+8F7EFEEF@EGGGEEE

</code></pre>

<p>Each sequencing read is represented by four lines:</p>

<ol>
<li>@ followed by read ID and optional information about sequencing run</li>
<li>sequenced bases</li>
<li>+ (optionally followed by the read ID and some additional info)</li>
<li>quality scores for each base of the sequence encoded as <a href="https://en.wikipedia.org/wiki/ASCII">ASCII symbols</a></li>
</ol>

<h2 id="paired-end-data">Paired end data</h2>

<p>It is common to prepare pair-end and mate-pair sequencing libraries. This is highly beneficial for a number of applications discussed in subsequent topics. For now let&rsquo;s just briefly discuss what these are and how they manifest themselves in fastq form.</p>

<blockquote>

<figure >
    
        <img src="/BMMB554/img/pe_mp.png" />
    
    
</figure>


<p><strong>Paired-end and mate-pair reads</strong>. In paired end sequencing (left) the actual ends of rather short DNA molecules (&lt;1kb) are determined, while for mate pair sequencing (right) the ends of long molecules are joined and prepared in special sequencing libraries. In these mate pair protocols, the ends of long, size-selected molecules are connected with an internal adapter sequence (i.e. linker, yellow) in a circularization reaction. The circular molecule is then processed using restriction enzymes or fragmentation. Fragments are enriched for the linker and outer library adapters are added around the two combined molecule ends. The internal adapter can then be used as a second priming site for an additional sequencing reaction in the same orientation or sequencing can be performed from the second adapter, from the reverse strand. (From Ph.D. dissertation by <a href="https://core.ac.uk/download/pdf/35186947.pdf">Martin Kircher</a>)</p>
</blockquote>

<p>Thus in both cases (paired-end and mate-pair) a single physical piece of DNA (or RNA in the case of RNA-seq) is sequenced from two ends and so generates two reads. These can be represented as separate files (two fastq files with first and second reads) or a single file were reads for each end are interleaved. Here are examples:</p>

<h4 id="two-single-files">Two single files</h4>

<p><strong>File 1</strong></p>

<pre><code> @M02286:19:000000000-AA549:1:1101:12677:1273 1:N:0:23
 CCTACGGGTGGCAGCAGTGAGGAATATTGGTCAATGGACGGAAGTCT
 +
 ABC8C,:@F:CE8,B-,C,-6-9-C,CE9-CC--C-&lt;-C++,,+;CE
 @M02286:19:000000000-AA549:1:1101:15048:1299 1:N:0:23
 CCTACGGGTGGCTGCAGTGAGGAATATTGGACAATGGTCGGAAGACT
 +
 ABC@CC77CFCEG;F9&lt;F89&lt;9--C,CE,--C-6C-,CE:++7:,CF
</code></pre>

<hr />

<p><strong>File 2</strong></p>

<pre><code>@M02286:19:000000000-AA549:1:1101:12677:1273 2:N:0:23
CACTACCCGTGTATCTAATCCTGTTTGATACCCGCACCTTCGAGCTTA
+
--8A,CCE+,,;,&lt;CC,,&lt;CE@,CFD,,C,CFF+@+@CCEF,,,B+C,
@M02286:19:000000000-AA549:1:1101:15048:1299 2:N:0:23
CACTACCGGGGTATCTAATCCTGTTCGCTCCCCACGCTTTCGTCCATC
+
-6AC,EE@::CF7CFF&lt;&lt;FFGGDFFF,@FGGGG?F7FEGGGDEFF&gt;FF
</code></pre>

<hr />

<p>Note that read ID are <strong>identical</strong> in two files and they are listed in <strong>the same</strong> order. In some cases read IDs in the first and second file may be appended with <code>/1</code> and <code>/2</code> tags, respectively.</p>

<h4 id="interleaved-file">Interleaved file</h4>

<pre><code>@1/1
AGGGATGTGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTA
+
EGGEGGGDFGEEEAEECGDEGGFEEGEFGBEEDDECFEFDD@CDD&lt;ED
@1/2
CCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAAC
+
GHHHDFDFGFGEGFBGEGGEGEGGGHGFGHFHFHHHHHHHEF?EFEFF
@2/1
AGGGATGTGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTA
+
HHHHHHEGFHEEFEEHEEHHGGEGGGGEFGFGGGGHHHHFBEEEEEFG
@2/2
CCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAAC
+
HHHHHHHHHHHHHGHHHHHHGHHHHHHHHHHHFHHHFHHHHHHHHHHH
</code></pre>

<hr />

<p>Here the first and the second reads are identified with <code>/1</code> and <code>/2</code> tags.</p>

<p><font color="red">&#10148;</font> <strong>Note</strong>: Fastq format is not strictly defined and its variations will always cause headache for you. See <a href="https://www.ncbi.nlm.nih.gov/books/NBK242622/">this page</a> for more information.</p>

<h2 id="what-are-base-qualities">What are base qualities?</h2>

<p>As we&rsquo;ve seen above, fastq datasets contain two types of information:</p>

<ul>
<li><em>sequence of the read</em></li>
<li><em>base qualities</em> for each nucleotide in the read.</li>
</ul>

<p>The base qualities allow us to judge how trustworthy each base in a sequencing read is. The following excerpt from an excellent <a href="http://chagall.med.cornell.edu/RNASEQcourse/Intro2RNAseq.pdf">tutorial</a> by Friederike D&uuml;ndar, Luce Skrabanek, Paul Zumbo explains what base qualities are:</p>

<p>Illumina sequencing is based on identifying the individual nucleotides by the fluorescence signal emitted upon their incorporation into the growing sequencing read. Once the fluorescence intensities are extracted and translated into the four letter code. The deduction of nucleotide sequences from the images acquired during sequencing is commonly referred to as base calling. Due to the imperfect nature of the sequencing process and limitations of the optical instruments, base calling will always have inherent uncertainty. This is the reason why FASTQ files store the DNA sequence of each read together with a position-specific quality score that represents the error probability, i.e., how likely it is that an individual base call may be incorrect. The score is called <a href="http://www.phrap.com/phred/">Phred score</a>, $Q$, which is proportional to the probability $p$ that a base call is incorrect, where $Q = −10lg(p)$. For example, a Phred score of 10 corresponds to one error in every ten base calls ($Q = −10lg(0.1)$), or 90% accuracy; a Phred score of 20 corresponds to one error in every 100 base calls, or 99% accuracy. A higher Phred score thus reflects higher confidence in the reported base. To assign each base a unique score identifier (instead of numbers of varying character length), Phred scores are typically represented as ASCII characters. At <a href="http://ascii-code.com/">http://ascii-code.com/</a> you can see which characters are assigned to what number. For raw reads, the range of scores will depend on the sequencing technology and the base caller used (Illumina, for example, used a tool called Bustard, or, more recently, RTA). Unfortunately, Illumina has been anything but consistent in how they a) calculated and b) ASCII-encoded the Phred score (see below)! In addition, Illumina now allows Phred scores for base calls with as high as 45, while 41 used to be the maximum score until the HiSeq X. This may cause issues with downstream sapplications that expect an upper limit of 41.</p>

<hr />


<figure >
    
        <img src="/BMMB554/img/illumina_qs.png" />
    
    
</figure>


<p>Base call quality scores are represented with the Phred range. Different Illumina (formerly Solexa) versions
used different scores and ASCII offsets. Starting with Illumina format 1.8, the score now represents the standard
Sanger/Phred format that is also used by other sequencing platforms and the sequencing archives.</p>

<hr />


<figure >
    
        <img src="/BMMB554/img/fastq_qs.png" />
    
    
</figure>


<p>The ASCII interpretation and ranges of the different Phred score notations used by Illumina and the original
Sanger interpretation. Although the Sanger format allows a theoretical score of 93, raw sequencing
reads typically do not exceed a Phred score of 60. In fact, most Illumina-based sequencing will result in maximum
scores of 41 to 45 (image from <a href="https://en.wikipedia.org/wiki/FASTQ_format">Wikipedia</a>).</p>

<h2 id="assessing-data-quality">Assessing data quality</h2>

<p>One of the first steps in the analysis of NGS data is seeing how good the data actually is. <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastqQC</a> is a fantastic tool allowing you to gauge the quality of fastq datasets (and deciding whether to blame or not to blame whoever has done sequencing for you).</p>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="left"></th>
</tr>
</thead>

<tbody>
<tr>
<td align="left"><img src="https://galaxyproject.org/ngs101/good_fq.png" alt="" /></td>
<td align="left"><img src="https://galaxyproject.org/ngs101/bad_fq.png" alt="" /></td>
</tr>

<tr>
<td align="left"><strong>A.</strong> Excellent quality</td>
<td align="left"><strong>B.</strong> Hmmm&hellip;OK</td>
</tr>
</tbody>
</table>

<p>Here you can see FastQC base quality reports (the tools gives you many other types of data) for two datasets: <strong>A</strong> and <strong>B</strong>. The <strong>A</strong> dataset has long reads (250 bp) and very good quality profile with no qualities dropping below <a href="http://www.phrap.com/phred/">phred score</a> of 30. The <strong>B</strong> dataset is significantly worse with ends of the reads dipping below phred score of 20. The <strong>B</strong> reads may need to be trimmed for further processing.</p>


<div style="position: relative; padding-bottom: 56.25%; padding-top: 30px; height: 0; overflow: hidden;">
  <iframe src="//player.vimeo.com/video/123453134" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
 </div>


<h3 id="font-color-red-try-it-yourself-font"><font color="red">Try it yourself</font></h3>

<p>QC datasets you have uploaded before.</p>

<h2 id="mapping-your-data">Mapping your data</h2>

<p>Mapping of NGS reads against reference sequences is one of the key steps of the analysis. In <a href="/post/topic5/">Topic 5</a> we covered basic principles behind mapping of sequencing reads against large genomes. Now it is time to see how this is done in practice. Below is a list of key publications highlighting popular mapping tools:</p>

<ul>
<li>2009 Bowtie 1 - <a href="http://genomebiology.com/content/10/3/R25">Langmead et al.</a></li>
<li>2012 Bowtie 2 - <a href="http://www.nature.com/nmeth/journal/v9/n4/full/nmeth.1923.htm">Langmead and Salzberg</a></li>
<li>2009 BWA - <a href="http://bioinformatics.oxfordjournals.org/content/25/14/1754.long">Li and Durbin</a></li>
<li>2010 BWA - <a href="http://bioinformatics.oxfordjournals.org/content/26/5/589">Li and Durbin</a></li>
<li>2013 BWA-MEM - <a href="http://arxiv.org/abs/1303.3997">Li</a></li>
</ul>

<h3 id="mapping-against-a-pre-computed-genome-index">Mapping against a pre-computed genome index</h3>

<p>Mappers usually compare reads against a reference sequence that has been transformed into a highly accessible data structure called genome index. Such indexes should be generated before mapping begins. Galaxy instances typically store indexes for a number of publicly available genome builds.</p>

<blockquote>

<figure >
    
        <img src="/BMMB554/img/cached_genome.png" />
    
    
</figure>


<p>Mapping against a pre-computed index in Galaxy</p>
</blockquote>

<p>For example, the image above shows indexes for <code>hg38</code> version of the human genome. You can see that there are actually three choices: (1) <code>hg38</code>, (2) <code>hg38 canonical</code> and (3) <code>hg38 canonical female</code>. The <code>hg38</code> contains all chromosomes as well as all unplaced contigs. The <code>hg38 canonical</code> does not contain unplaced sequences and only consists of chromosomes 1 through 22, X, Y, and mitochondria. The
<code>hg38 canonical female</code> contains everything from the canonical set with the exception of chromosome Y.</p>

<p>The following video show mapping using BWA:</p>


<div style="position: relative; padding-bottom: 56.25%; padding-top: 30px; height: 0; overflow: hidden;">
  <iframe src="//player.vimeo.com/video/123102338" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
 </div>


<h3 id="font-color-red-try-it-yourself-font-1"><font color="red">Try it yourself</font></h3>

<p>Map datasets uploaded before using BWA against <code>hg38</code> version of the human genome.</p>

<h3 id="what-if-pre-computed-index-does-not-exist">What if pre-computed index does not exist?</h3>

<p>If Galaxy does not have a genome you need to map against, you can upload your genome sequence as a FASTA file and use it in the mapper directly as shown below (<strong>Load reference genome</strong> is set to <code>History</code>).</p>

<blockquote>

<figure >
    
        <img src="/BMMB554/img/uploaded_genome.png" />
    
    
</figure>


<p>Mapping against a pre-computed index in Galaxy</p>
</blockquote>

<p>In this case Galaxy will first create an index from this dataset and then run mapping analysis against it. The following video shows how this works in practice:</p>


<div style="position: relative; padding-bottom: 56.25%; padding-top: 30px; height: 0; overflow: hidden;">
  <iframe src="//player.vimeo.com/video/123108417" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
 </div>


<h2 id="sam-bam-datasets">SAM/BAM datasets</h2>

<p>The <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">SAM/BAM</a> format is an accepted standard for storing aligned reads (it can also store unaligned reads and some mappers such as BWA are accepting unaligned BAM as input). The binary form of the format (BAM) is compact and can be rapidly searched (if indexed). In Galaxy BAM datasets are always indexed (accompanies by a .bai file) and sorted in coordinate order. In the following duscussion I once again rely on <a href="http://chagall.med.cornell.edu/RNASEQcourse/Intro2RNAseq.pdf">tutorial</a> by Friederike D&uuml;ndar, Luce Skrabanek, and Paul Zumbo.</p>

<p>The output option of STAR already indicates that the results of the alignment will be stored in a SAM or BAM file. The Sequence Alignment/Map (SAM) format is, in fact, a generic nucleotide alignment format that describes the alignment of sequencing reads (or query sequences) to a reference. The human readable, TABdelimited SAM files can be compressed into the Binary Alignment/Map format. These BAM files are bigger than simply gzipped SAM files, because they have been optimized for fast random access rather than size reduction. Position-sorted BAM files can be indexed so that all reads aligning to a locus can be efficiently retrieved without loading the entire file into memory.</p>

<p>As shown below, SAM files typically contain a short header section and a very long alignment section where each row represents a single read alignment. The following sections will explain the SAM format in a bit more detail. For the most comprehensive and updated information go to <a href="https://github.com/samtools/hts-specs">https://github.com/samtools/hts-specs</a>.</p>


<figure >
    
        <img src="/BMMB554/img/bam_structure.png" />
    
    
</figure>


<p>Schematic representation of a SAM file. Each line of the optional header section starts with “@”, followed by the appropriate abbreviation (e.g., SQ for sequence dictionary which lists all chromosomes names (SN) and their lengths (LN)). The vast majority of lines within a SAM file typically correspond to read alignments where each read is described by the 11 mandatory entries (black font) and a variable number of optional fields (grey font).</p>

<h3 id="sam-header">SAM Header</h3>

<p>The header section includes information about how the alignment was generated and stored. All lines in the header section are tab-delimited and begin with the “@” character, followed by tag:value pairs, where tag is a two-letter string that defines the content and the format of value. For example, the “@SQ” line in the header section contains the information about the names and lengths of the *reference sequences to which the reads were aligned. For a hypothetical organism with three chromosomes of length 1,000 bp, the SAM header should contain the following three lines:</p>

<pre><code>@SQ SN:chr1 LN:1000
@SQ SN:chr2 LN:1000
@SQ SN:chr3 LN:1000
</code></pre>

<h3 id="sam-alignment-section">SAM alignment section</h3>

<p>The optional header section is followed by the alignment section where each line corresponds to one sequenced read. For each read, there are 11 mandatory fields that always appear in the same order:</p>

<pre><code>&lt;QNAME&gt; &lt;FLAG&gt; &lt;RNAME&gt; &lt;POS&gt; &lt;MAPQ&gt; &lt;CIGAR&gt; &lt;MRNM&gt; &lt;MPOS&gt; &lt;ISIZE&gt; &lt;SEQ&gt; &lt;QUAL&gt;
</code></pre>

<p>If the corresponding information is unavailable or irrelevant, field values can be ‘0’ or ‘*’ (depending on the field, see below), but they cannot be missing! After the 11 mandatory fields, a variable number of optional fields can be present. Here’s an example of one single line of a real-life SAM file (you may need to scroll sideways):</p>

<pre><code>ERR458493 .552967 16 chrI 140 255 12 M61232N37M2S * 0 0 CCACTCGTTCACCAGGGCCGGCGGGCTGATCACTTTATCGTGCATCTTGGC BB?HHJJIGHHJIGIIJJIJGIJIJJIIIGHBJJJJJJHHHHFFDDDA1+B NH:i:1 HI:i:1 AS:i:41 nM:i:2
</code></pre>

<p>The following table explains the format and content of each field. The <code>FLAG</code>, <code>CIGAR</code>, and the optional fields (marked in blue) are explained in more detail below. The number of optional fields can vary widely between different SAM files and even between reads within in the same file. The field types marked in blue are explained in more detail in the main text below.</p>


<figure >
    
        <img src="/BMMB554/img/sam_fields.png" />
    
    
</figure>


<h4 id="flag-field"><code>FLAG</code> field</h4>

<p>The FLAG field encodes various pieces of information about the individual read, which is particularly important for PE reads. It contains an integer that is generated from a sequence of Boolean bits (0, 1). This way, answers to multiple binary (Yes/No) questions can be compactly stored as a series of bits, where each of the single bits can be addressed and assigned separately.</p>

<p>The following table gives an overview of the different properties that can be encoded in the FLAG field. The developers of the SAM format and samtools tend to use the hexadecimal encoding as a means to refer to the different bits in their documentation. The value of the FLAG field in a given SAM file, however, will always be the decimal representation of the sum of the underlying binary values (as shown in Table below, row 2).</p>


<figure >
    
        <img src="/BMMB554/img/sam_flag.png" />
    
    
</figure>


<p>The <code>FLAG</code> field of SAM files stores information about the respective read alignment in one single decimal number. The decimal number is the sum of all the answers to the Yes/No questions associated with each binary bit. The hexadecimal representation is used to refer to the individual bits (questions). A bit is set if the corresponding state is true. For example, if a read is paired, <code>0x1</code> will be set, returning the decimal value of 1. Therefore, all <code>FLAG</code> values associated with paired reads must be uneven decimal numbers. Conversely, if the <code>0x1</code> bit is unset (= read is not paired), no assumptions can be made about <code>0x2</code>, <code>0x8</code>, <code>0x20</code>, <code>0x40</code> and <code>0x80</code> because they refer to paired reads.</p>

<p>In a run with single reads, the flags you most commonly see are:</p>

<ul>
<li>0: This read has been mapped to the forward strand. (None of the bit-wise flags have been set.)</li>
<li>4: The read is unmapped (<code>0x4</code> is set).</li>
<li>16: The read is mapped to the reverse strand (<code>0x10</code> is set)</li>
</ul>

<p>(<code>0x100</code>, <code>0x200</code> and <code>0x400</code> are not used by most aligners/mappers, but could, in principle be set for single reads.) Some common <code>FLAG</code> values that you may see in a PE experiment include:</p>

<table>
<thead>
<tr>
<th></th>
<th></th>
</tr>
</thead>

<tbody>
<tr>
<td><strong>69</strong> (= 1 + 4 + 64)</td>
<td>The read is paired, is the first read in the pair, and is unmapped.</td>
</tr>

<tr>
<td><strong>77</strong> (= 1 + 4 + 8 + 64)</td>
<td>The read is paired, is the first read in the pair, both are unmapped.</td>
</tr>

<tr>
<td><strong>83</strong> (= 1 + 2 + 16 + 64)</td>
<td>The read is paired, mapped in a proper pair, is the first read in the pair, and it is mapped to the reverse strand.</td>
</tr>

<tr>
<td><strong>99</strong> (= 1 + 2 + 32 + 64)</td>
<td>The read is paired, mapped in a proper pair, is the first read in the pair, and its mate is mapped to the reverse strand.</td>
</tr>

<tr>
<td><strong>133</strong> (= 1 + 4 + 128)</td>
<td>The read is paired, is the second read in the pair, and it is unmapped.</td>
</tr>

<tr>
<td><strong>137</strong> (= 1 + 8 + 128)</td>
<td>The read is paired, is the second read in the pair, and it is mapped while its mate is not.</td>
</tr>

<tr>
<td><strong>141</strong> (= 1 + 4 + 8 + 128)</td>
<td>The read is paired, is the second read in the pair, but both are unmapped.</td>
</tr>

<tr>
<td><strong>147</strong> (= 1 + 2 + 16 + 128)</td>
<td>The read is paired, mapped in a proper pair, is the second read in the pair, and mapped to the reverse strand.</td>
</tr>

<tr>
<td><strong>163</strong> (= 1 + 2 + 32 + 128)</td>
<td>The read is paired, mapped in a proper pair, is the second read in the pair, and its mate is mapped to the reverse strand.</td>
</tr>
</tbody>
</table>

<p>A useful website for quickly translating the FLAG integers into plain English explanations like the ones shown above is: <a href="https://broadinstitute.github.io/picard/explain-flags.html">https://broadinstitute.github.io/picard/explain-flags.html</a></p>

<h4 id="cigar-string"><code>CIGAR</code> string</h4>

<p><code>CIGAR</code> stands for <em>Concise Idiosyncratic Gapped Alignment Report</em>. This sixth field of a SAM file
contains a so-called CIGAR string indicating which operations were necessary to map the read to the reference sequence at that particular locus.</p>

<p>The following operations are defined in CIGAR format (also see figure below):</p>

<ul>
<li><strong>M</strong> - Alignment (can be a sequence match or mismatch!)</li>
<li><strong>I</strong> - Insertion in the read compared to the reference</li>
<li><strong>D</strong> - Deletion in the read compared to the reference</li>
<li><strong>N</strong> - Skipped region from the reference. For mRNA-to-genome alignments, an N operation represents an intron. For other types of alignments, the interpretation of N is not defined.</li>
<li><strong>S</strong> - Soft clipping (clipped sequences are present in read); S may only have H operations between them and the ends of the string</li>
<li><strong>H</strong> - Hard clipping (clipped sequences are NOT present in the alignment record); can only be present as the first and/or last operation</li>
<li><strong>P</strong> - Padding (silent deletion from padded reference)</li>
<li><strong>=</strong> - Sequence match (not widely used)</li>
<li><strong>X</strong> - Sequence mismatch (not widely used)</li>
</ul>

<p>The sum of lengths of the <strong>M</strong>, <strong>I</strong>, <strong>S</strong>, <strong>=</strong>, <strong>X</strong> operations must equal the length of the read. Here are some examples:</p>


<figure >
    
        <img src="/BMMB554/img/cigar.png" />
    
    
</figure>


<h4 id="optional-fields">Optional fields</h4>

<p>Following the eleven mandatory SAM file fields, the optional fields are presented as key-value
pairs in the format of <code>&lt;TAG&gt;:&lt;TYPE&gt;:&lt;VALUE&gt;</code>, where <code>TYPE</code> is one of:</p>

<ul>
<li><code>A</code> - Character</li>
<li><code>i</code> - Integer</li>
<li><code>f</code> - Float number</li>
<li><code>Z</code> - String</li>
<li><code>H</code> - Hex string</li>
</ul>

<p>The information stored in these optional fields will vary widely depending on the mapper and new tags can be added freely. In addition, reads within the same SAM file may have different numbers of optional fields, depending on the program that generated the SAM file. Commonly used optional tags include:</p>

<ul>
<li><code>AS:i</code> - Alignment score</li>
<li><code>BC:Z</code> - Barcode sequence</li>
<li><code>HI:i</code> - Match is i-th hit to the read</li>
<li><code>NH:i</code> - Number of reported alignments for the query sequence</li>
<li><code>NM:i</code> - Edit distance of the query to the reference</li>
<li><code>MD:Z</code> - String that contains the exact positions of mismatches (should complement the CIGAR string)</li>
<li><code>RG:Z</code> - Read group (should match the entry after ID if @RG is present in the header.</li>
</ul>

<p>Thus, for example, we can use the NM:i:0 tag to select only those reads which map perfectly to the reference(i.e., have no mismatches). While the optional fields listed above are fairly standardized, tags that begin with <code>X</code>, <code>Y</code>, and <code>Z</code> are reserved for particularly free usage and will never be part of the official SAM file format specifications. <code>XS</code>, for example, is used by TopHat (an RNA-seq analysis tool we will discuss later) to encode the strand information (e.g., <code>XS:A:+</code>) while Bowtie2 and BWA use <code>XS:i:</code> for reads with multiple alignments to store the alignment score for the next-best-scoring alignment (e.g., <code>XS:i:30</code>).</p>

<h3 id="read-groups">Read Groups</h3>

<p>One of the key features of SAM/BAM format is the ability to label individual reads with readgroup tags. This allows pooling results of multiple experiments into a single BAM dataset. This significantly simplifies downstream logistics: instead of dealing with multiple datasets one can handle just one. Many downstream analysis tools such as variant callers are designed to recognize readgroup data and output results on per-readgroup basis.</p>

<p>One of the best descriptions of BAM readgroups is on <a href="http://gatkforums.broadinstitute.org/discussion/1317/collected-faqs-about-bam-files">GATK support site</a>. We have gratefully stolen two tables describing the most important readgroup tags - <code>ID</code>, <code>SM</code>, <code>LB</code>, and <code>PL</code> - from GATK forum and provide them here:</p>


<figure >
    
        <img src="/BMMB554/img/rg.png" />
    
    
</figure>


<p>As further described in the GATK forum:&rdquo;<em>A concrete example may be instructive. Suppose I have a trio of samples: MOM, DAD, and KID. Each has two DNA libraries prepared, one with 400 bp inserts and another with 200 bp inserts. Each of these libraries is run on two lanes of an Illumina machine, requiring 3 x 2 x 2 = 12 lanes of data. When the data come off the sequencer, I would create 12 bam files, with the following @RG fields in the header</em>&ldquo;:</p>


<figure >
    
        <img src="/BMMB554/img/rg_example.png" />
    
    
</figure>


<p>Too see an example of read group manipulation in Galaxy see the following <a href="https://player.vimeo.com/video/123102338#t=1:40">clip</a>.</p>

<h2 id="manipulating-sam-bam-datasets">Manipulating SAM/BAM datasets</h2>

<p>We support four major toolsets for processing of SAM/BAM datasets:</p>

<ul>
<li><a href="https://deeptools.github.io/">DeepTools</a> - a suite of user-friendly tools for the visualization, quality control and normalization of data from deep-sequencing DNA sequencing experiments.</li>
<li><a href="[http://www.htslib.org/">SAMtools</a> - various utilities for manipulating alignments in the SAM/BAM format, including sorting, merging, indexing and generating alignments in a per-position format.</li>
<li><a href="https://github.com/pezmaster31/bamtools/wiki/Tutorial_Toolkit_BamTools-1.0.pdf">BAMtools</a> - a toolkit for reading, writing, and manipulating BAM (genome alignment) files.</li>
<li><a href="http://broadinstitute.github.io/picard/">Picard</a> - a set of Java tools for manipulating high-throughput sequencing data (HTS) data and formats.</li>
</ul>

<p>The following video highlights de-duplication, filtering, and cleaning of a BAM dataset using BAMtools and Picard tools:</p>


<div style="position: relative; padding-bottom: 56.25%; padding-top: 30px; height: 0; overflow: hidden;">
  <iframe src="//player.vimeo.com/video/123113197" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
 </div>


<h3 id="font-color-red-try-it-yourself-font-2"><font color="red">Try it yourself</font></h3>

<p>Perform a similar analyses with your own data.</p>

<h2 id="the-challenge-of-read-duplicates">The challenge of read duplicates</h2>

<h3 id="pcr-duplicates">PCR duplicates</h3>

<p>Preparation of sequencing libraries (at least at the time of writing) for technologies such as Illumina (used in this examples) involves PCR amplification. It is required to generate sufficient number of sequencing templates so that a reliable detection can be performed by base callers. Yet PCR has it&rsquo;s biases, which are especially profound in cases of multitemplate PCR used for construction of sequencing libraries (Kanagawa et al. <a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=16233530">2003</a>).</p>

<blockquote>
<p>
<figure >
    
        <img src="/BMMB554/img/pcr-duplicates.png" />
    
    
</figure>

Analyzing molecules aligning with the same outer coordinates, a mapping quality of at least 30 and a length of at least 30nt, resulted in an average coverage of 12.9 per PCR duplicate and an empirical coverage distribution similar to an exponential/power law distribution (left upper panel). This indicates that many molecules are only observed for deeper sequencing while other molecules are available at higher frequencies. Analyzing length (left middle panel) and GC content (left lower panel) patterns as well as the combination (right panel) shows higher PCR duplicate counts for a GC content between 30% to 70% as well as for shorter molecules compared to longer molecules. This effect may be due to an amplification bias from the polymerase or the cluster generation process necessary for Illumina sequencing. From Ph.D. dissertation of <a href="http://www.qucosa.de/fileadmin/data/qucosa/documents/7110/pflichtexemplar_final.pdf">Martin Kircher</a>).</p>
</blockquote>

<p>Duplicates can be identified based on their outer alignment coordinates or using sequence-based clustering. One of the common ways for identification of duplicate reads is the <code>MarkDuplicates</code> utility from <a href="https://broadinstitute.github.io/picard/command-line-overview.html">Picard</a> package. It is designed to identify both PCR and optical duplicates:</p>

<blockquote>
<p>Duplicates are identified as read pairs having identical 5&rsquo; positions (coordinate and strand) for both reads in a mate pair (and optionally, matching unique molecular identifier reads; see BARCODE_TAG option). Optical, or more broadly Sequencing, duplicates are duplicates that appear clustered together spatially during sequencing and can arise from optical/imagine-processing artifacts or from bio-chemical processes during clonal amplification and sequencing; they are identified using the READ_NAME_REGEX and the OPTICAL_DUPLICATE_PIXEL_DISTANCE options. The tool&rsquo;s main output is a new SAM or BAM file in which duplicates have been identified in the SAM flags field, or optionally removed (see REMOVE_DUPLICATE and REMOVE_SEQUENCING_DUPLICATES), and optionally marked with a duplicate type in the &lsquo;DT&rsquo; optional attribute. In addition, it also outputs a metrics file containing the numbers of READ_PAIRS_EXAMINED, UNMAPPED_READS, UNPAIRED_READS, UNPAIRED_READ DUPLICATES, READ_PAIR_DUPLICATES, and READ_PAIR_OPTICAL_DUPLICATES. Usage example: java -jar picard.jar MarkDuplicates I=input.bam \ O=marked_duplicates.bam M=marked_dup_metrics.txt.`</p>
</blockquote>

<h3 id="sampling-coincidence-duplicates">Sampling coincidence duplicates</h3>

<p>However, one has to be careful when removing duplicates in cases when the sequencing targets are small (e.g., sequencing of bacterial, viral, or organellar genomes as well as amplicons). This is because when sequencing target is small reads will have the same coordinates by chance and not because of PCR amplification issues. The figure below illustrates the fine balance between estimates allele frequency, coverage, and variation in insert size:</p>

<blockquote>
<p>
<figure >
    
        <img src="/BMMB554/img/sampling-bias.png" />
    
    
</figure>

The Variant Allele Frequency (VAF) bias determined by coverage and insert size variance. Reads are paired-end and read length is 76. The insert size distribution is modeled as a Gaussian distribution with mean at 200 and standard deviation shown on the x-axis. The true VAF is 0.05. The darkness at each position indicates the magnitude of the bias in the VAF. (From Zhou et al. <a href="http://bioinformatics.oxfordjournals.org/content/30/8/1073">2013</a>).</p>
</blockquote>

    </section>

    <hr>

    <footer class="post-footer">
        <section class="f-1">
            
            
            <p class="f-post-time"><time datetime="2017-02-01T11:26:49-04:00">February 1, 2017</time></p>
        </section>
                        
        <section class="f-2">
            <section class="share">
                <span>Share:
                <a class="icon-twitter" href="http://twitter.com/share?text=7.%20NGS%20data%3a%20Practicalities&url=https%3a%2f%2fnekrut.github.io%2fBMMB554%2fpost%2ftopic7%2f"
                    onclick="window.open(this.href, 'twitter-share', 'width=550,height=235');return false;">
                    <i class="fa fa-twitter"></i>
                </a>
                <a class="icon-facebook" href="https://www.facebook.com/sharer/sharer.php?u=https%3a%2f%2fnekrut.github.io%2fBMMB554%2fpost%2ftopic7%2f"
                    onclick="window.open(this.href, 'facebook-share','width=580,height=296');return false;">
                    <i class="fa fa-facebook"></i>
                </a>
                <a class="icon-google-plus" href="https://plus.google.com/share?url=https%3a%2f%2fnekrut.github.io%2fBMMB554%2fpost%2ftopic7%2f"
                   onclick="window.open(this.href, 'google-plus-share', 'width=490,height=530');return false;">
                    <i class="fa fa-google-plus"></i>
                </a>
                </span>
            </section>

            
        </section>
                        
    </footer>
</article>
	</div>
</main>
    <footer id="site-footer">
    <hr>
        <div class="container">
            
            <section><a href="https://nekrut.github.io/BMMB554/">Anton Nekrutenko</a> 2017 | MIT License</section>
        </div>
    </footer>

    <script type="text/javascript" src="https://nekrut.github.io/BMMB554/js/fittext.js"></script>
    <script type="text/javascript">
      $(".heading").fitText();
    </script>
    <script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/x-mathjax-config">
  MathJax.Hub.Queue(function() {
    
    
    
    var all = MathJax.Hub.getAllJax(), i;
    for(i = 0; i < all.length; i += 1) {
        all[i].SourceElement().parentNode.className += ' has-jax';
    }
});
</script>


</body>
</html>