<!DOCTYPE html>
<!--[if lt IE 7]><html class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
<!--[if IE 7]><html class="no-js lt-ie9 lt-ie8"> <![endif]-->
<!--[if IE 8]><html class="no-js lt-ie9"> <![endif]-->
<!--[if gt IE 8]><!--><html class="no-js" prefix="og: http://ogp.me/ns#" xmlns:og="http://ogp.me/ns#"><!--<![endif]-->

    <head>
                <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="description" content="**Topic 10** | Why calling variants?">
        <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=1.0" />
        <meta name="mobile-web-app-capable" content="yes">
        <meta name="apple-mobile-web-app-capable" content="yes">
        <meta name="apple-mobile-web-app-status-bar-style" content="black">
        <meta name="author" content="Anton Nekrutenko">
  
	
        <meta property="og:site_name" content="BMMB 554 / 2017">
        <meta property="og:title" content="BMMB 554 / 2017">
        <meta property="og:url" content="https://nekrut.github.io/BMMB554/post/topic9a/">
        <meta property="og:description" content="Introduction to data driven life sciences">
    
        <meta property="og:type" content="article" />
        <meta property="og:article:author" content="Anton Nekrutenko" />
        <meta property="og:article:published_time" content="2017-02-20T10:11:10-04:00" />
    
        <meta name="generator" content="Hugo 0.19" />
        <title>10. Why calling variants? &middot; BMMB 554 / 2017</title>
        <link rel="canonical" href="https://nekrut.github.io/BMMB554/" />
        <link rel="alternate" type="application/rss+xml" title="RSS" href="">
        <link rel="stylesheet" type="text/css" href="https://nekrut.github.io/BMMB554/css/main.css"/>
        <link href="//fonts.googleapis.com/css?family=Source+Sans+Pro:200|Montserrat:700" rel="stylesheet" type="text/css">
        <link href="//netdna.bootstrapcdn.com/font-awesome/4.0.3/css/font-awesome.css" rel="stylesheet">
        <script src="//code.jquery.com/jquery-1.10.2.min.js"></script>
    </head>

<body>
<!--[if lt IE 7]><p class="browsehappy">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> or <a href="http://www.google.com/chrome/‎">install Google Chrome</a> to experience this site.</p><![endif]-->

    <header id="site-header">
        <div class="container">
            <a href="https://nekrut.github.io/BMMB554/" alt="BMMB 554 / 2017"><h1 class="blog-title heading">BMMB 554 / 2017</h1></a>
            <p class="blog-description">Introduction to data driven life sciences</p>
        </div>
    </header>
<main class="content" role="main">
	<div class="container">
		<article class="post">
	<header class="post-header">
        <h3 class="p-post-title">10. Why calling variants?</h3>
    </header>

    <section class="post-content">
        

<h1 id="tibet-is-high">Tibet is high</h1>

<p>Many residents of the Tibetian Plateau live above 4,000 meters where oxygen concentration is approximately 40% lower than at sea level:</p>

<blockquote>
<p><img src="/BMMB554/img/tibet.png" alt="" /></p>

<p>Tibetan Plateau and surrounding areas (from <a href="https://en.wikipedia.org/wiki/Tibet">Wikipedia</a>).</p>
</blockquote>

<p>Tibetian experience a number of adaptations to high altitudes manifesting in the following phenotypic differences:</p>

<ul>
<li>lower hemoglobin concentration <a href="http://science.sciencemag.org/lookup/ijlink?linkType=ABST&amp;journalCode=jap&amp;resid=98/2/598&amp;atom=%2Fsci%2F329%2F5987%2F75.atom">Wu:2005</a></li>
<li>higher arterial oxygen saturation <a href="http://science.sciencemag.org/lookup/ijlink?linkType=ABST&amp;journalCode=jap&amp;resid=98/2/598&amp;atom=%2Fsci%2F329%2F5987%2F75.atom">Niermeyer:1995</a></li>
<li>more efficient pulmonary gas exchange <a href="https://www.ncbi.nlm.nih.gov/pubmed/8822225">Zhuang:1996</a></li>
</ul>

<p>What are the genetic causes of these adaptations?</p>

<h1 id="sequencing-of-50-human-exomes-reveals-adaptation-to-high-altitude">Sequencing of 50 Human Exomes Reveals Adaptation to High Altitude</h1>

<p><a href="http://science.sciencemag.org/content/329/5987/75">Emilia Huerta-Sanchez and Rasmus Nielsen</a> decided to investigate this phenomenon using the following experimental design:</p>

<h2 id="design">Design</h2>

<p>They sequenced exomes from 50 unrelated individuals from two villages (both situated at or above 4,300 m above sea level) in the Tibet Autonomous Region of China. Exomes were enriched using <a href="http://sequencing.roche.com/products/nimblegen-seqcap-target-enrichment.html">Numblegen exome capture system</a> (approximately 20,000 genes) and sequenced to mean depth of ~18x.</p>

<h2 id="snps">SNPs</h2>

<p>18x coverage does not guarantee accurate assignment of individual genotypes. To address this challenge Nielsen et al. devised a Bayesian approach that would (1) estimate reliability for SNP calls and (2) compute population allele frequencies for each site.</p>

<p>They found:</p>

<ul>
<li>151,825 SNPs &gt; 50% of probability being variable</li>
<li>101,668 SNPs &gt; 99% probability of being variable</li>
</ul>

<p>Sanger sequencing validated 53 out of 56 that had &gt;95% probability of being variable and minor allele frequencies between 3% and 50%.</p>

<h2 id="population-history-of-tibetians-and-han-chinese">Population history of Tibetians and Han Chinese</h2>

<p>The Tibetian exome was compared with 40 Han Chinese from Beijing genomes (HCB) from the 1000 Genomes Project. Beijing is approximately 50 m above sea level and the adsolute Majority of Han Chinese live below 2,000 m. The amount of genetic differentiation between Tibetian and Han samples is low (<a href="https://en.wikipedia.org/wiki/Fixation_index">F<sub><em>ST</em></sub></a> = 0.026). In addition, there is also no differentiation between the two Tibetian villages (F<sub><em>ST</em></sub>=0.014) and so they are treated as a single population. The paper estimated that the Tibetian and Han populations diverged ~2,750 years ago with Tibetian population shrinking and Han population expanding.</p>

<blockquote>
<p><img src="https://d2ufo47lrtsv5s.cloudfront.net/content/sci/329/5987/75/F1.large.jpg?width=800&amp;height=600&amp;carousel=1" alt="" /></p>

<p>Two-dimensional unfolded site frequency spectrum for SNPs in Tibetan (x axis) and Han (y axis) population samples. The number of SNPs detected is color-coded according to the logarithmic scale plotted on the right. Arrows indicate a pair of intronic SNPs from the EPAS1 gene that show strongly elevated derived allele frequencies in the Tibetan sample compared with the Han sample.</p>
</blockquote>

<h2 id="finding-casual-snps">Finding casual SNPs</h2>

<p>With two populations we one can potentially find genes with significant allele frequency differences but we will not be able to tell which of the two populations is affected by selection. To do this we need an outgroup represented by a distantly related population. In this study they&rsquo;ve chosen a Danish population as an outgroup.</p>

<blockquote>
<p><img src="https://d2ufo47lrtsv5s.cloudfront.net/content/sci/329/5987/75/F2.large.jpg?width=800&amp;height=600&amp;carousel=1" alt="" /></p>

<p>Population-specific allele frequency change. (A) The distribution of F<sub><em>ST</em></sub>-based PBS statistics for the Tibetan branches, according to the number of variable sites in each gene. Outlier genes are indicated in red. (B) The signal of selection on EPAS1: Genomic average F<sub><em>ST</em></sub>-based branch lengths for Tibetan (T), Han (H), and Danish (D) branches (left) and branch lengths for EPAS1, indicating substantial differentiation along the Tibetan lineage (right).</p>
</blockquote>

<h2 id="epas1"><em>EPAS1</em></h2>

<p>The strongest signal was observed for the <em>EPAS1</em> gene, which has 9% derived allele frequency in Han Chinese and 87% in Tibetian population. To establish functional link between this site and inhabiting high altitudes authors performed association analysis:  &ldquo;<em>Significant associations were discovered for erythrocyte count (F test P = 0.00141) and for hemoglobin concentration (F test P = 0.00131), with significant or marginally significant P values for both traits when each village was tested separately
The allele at high frequency in the Tibetan sample was associated with lower erythrocyte quantities and correspondingly lower hemoglobin levels (table S4). Because elevated erythrocyte production is a common response to hypoxic stress, it may be that carriers of the “Tibetan” allele of EPAS1 are able to maintain sufficient oxygenation of tissues at high altitude without the need for increased erythrocyte levels. Thus, the hematological differences observed here may not represent the phenotypic target of selection and could instead reflect a side effect of EPAS1-mediated adaptation to hypoxic conditions.</em>&ldquo;</p>

<h1 id="altitude-adaptation-in-tibetans-caused-by-introgression-of-denisovan-like-dna">Altitude adaptation in Tibetans caused by introgression of Denisovan-like DNA</h1>

<p><a href="http://www.nature.com/nature/journal/v512/n7513/full/nature13408.html">Emilia Huerta-Sanchez and Rasmus Nielsen</a></p>

<blockquote>
<p><img src="http://www.nature.com/nature/journal/v512/n7513/images/nature13408-f2.jpg" alt="" /></p>

<p>Each column is a polymorphic genomic location (95 in total), each row is a phased haplotype (80 Han and 80 Tibetan haplotypes), and the coloured column on the left denotes the population identity of the individuals. Haplotypes of the Denisovan individual are shown in the top two rows (green). The black cells represent the presence of the derived allele and the grey space represents the presence of the ancestral allele (see Methods). The first and last columns correspond to the first and last positions in Supplementary Table 3, respectively. The red and blue arrows indicate the 32 sites in Supplementary Table 3. The blue arrows represent a five-SNP haplotype block defined by the first five SNPs in the 32.7-kb region. Asterisks indicate sites at which Tibetans share a derived allele with the Denisovan individual.</p>

<p><img src="http://www.nature.com/nature/journal/v512/n7513/images/nature13408-f3.jpg" alt="" /></p>

<p>The haplotypes were defined from all the SNPs present in the combined 1000 Genomes and Tibetan samples: 515 SNPs in total within the 32.7-kb EPAS1 region. The Denisovan haplotypes were added to the set of the common haplotypes. The R software package pegas23 was used to generate the figure, using pairwise differences as distances. Each pie chart represents one unique haplotype, labelled with Roman numerals, and the radius of the pie chart is proportional to the log2(number of chromosomes with that haplotype) plus a minimum size so that it is easier to see the Denisovan haplotype. The sections in the pie provide the breakdown of the haplotype representation amongst populations. The width of the edges is proportional to the number of pairwise differences between the joined haplotypes; the thinnest edge represents a difference of one mutation. The legend shows all the possible haplotypes among these populations. The numbers (1, 9, 35 and 40) next to an edge (the line connecting two haplotypes) in the bottom right are the number of pairwise differences between the corresponding haplotypes. We added an edge afterwards between the Tibetan haplotype XXXIII and its closest non-Denisovan haplotype (XXI) to indicate its divergence from the other modern human groups. Extended Data Fig. 5a contains all the pairwise differences between the haplotypes presented in this figure. ASW, African Americans from the south western United States; CEU, Utah residents with northern and western European ancestry; GBR, British; FIN, Finnish; JPT, Japanese; LWK, Luhya; CHS, southern Han Chinese; CHB, Han Chinese from Beijing; MXL, Mexican; PUR, Puerto Rican; CLM, Colombian; TSI, Toscani; YRI, Yoruban. Where there is only one line within a pie chart, this indicates that only one population contains the haplotype.</p>
</blockquote>

    </section>

    <hr>

    <footer class="post-footer">
        <section class="f-1">
            
            
            <p class="f-post-time"><time datetime="2017-02-20T10:11:10-04:00">February 20, 2017</time></p>
        </section>
                        
        <section class="f-2">
            <section class="share">
                <span>Share:
                <a class="icon-twitter" href="http://twitter.com/share?text=10.%20Why%20calling%20variants%3f&url=https%3a%2f%2fnekrut.github.io%2fBMMB554%2fpost%2ftopic9a%2f"
                    onclick="window.open(this.href, 'twitter-share', 'width=550,height=235');return false;">
                    <i class="fa fa-twitter"></i>
                </a>
                <a class="icon-facebook" href="https://www.facebook.com/sharer/sharer.php?u=https%3a%2f%2fnekrut.github.io%2fBMMB554%2fpost%2ftopic9a%2f"
                    onclick="window.open(this.href, 'facebook-share','width=580,height=296');return false;">
                    <i class="fa fa-facebook"></i>
                </a>
                <a class="icon-google-plus" href="https://plus.google.com/share?url=https%3a%2f%2fnekrut.github.io%2fBMMB554%2fpost%2ftopic9a%2f"
                   onclick="window.open(this.href, 'google-plus-share', 'width=490,height=530');return false;">
                    <i class="fa fa-google-plus"></i>
                </a>
                </span>
            </section>

            
        </section>
                        
    </footer>
</article>
	</div>
</main>
    <footer id="site-footer">
    <hr>
        <div class="container">
            
            <section><a href="https://nekrut.github.io/BMMB554/">Anton Nekrutenko</a> 2017 | MIT License</section>
        </div>
    </footer>

    <script type="text/javascript" src="https://nekrut.github.io/BMMB554/js/fittext.js"></script>
    <script type="text/javascript">
      $(".heading").fitText();
    </script>
    <script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/x-mathjax-config">
  MathJax.Hub.Queue(function() {
    
    
    
    var all = MathJax.Hub.getAllJax(), i;
    for(i = 0; i < all.length; i += 1) {
        all[i].SourceElement().parentNode.className += ' has-jax';
    }
});
</script>


</body>
</html>