<!DOCTYPE html>
<!--[if lt IE 7]><html class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
<!--[if IE 7]><html class="no-js lt-ie9 lt-ie8"> <![endif]-->
<!--[if IE 8]><html class="no-js lt-ie9"> <![endif]-->
<!--[if gt IE 8]><!--><html class="no-js" prefix="og: http://ogp.me/ns#" xmlns:og="http://ogp.me/ns#"><!--<![endif]-->

    <head>
                <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="description" content="**Topic 6** | Introduction to Galaxy platform for data manipulation and analysis">
        <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=1.0" />
        <meta name="mobile-web-app-capable" content="yes">
        <meta name="apple-mobile-web-app-capable" content="yes">
        <meta name="apple-mobile-web-app-status-bar-style" content="black">
        <meta name="author" content="Anton Nekrutenko">
  
	
        <meta property="og:site_name" content="BMMB 554 / 2017">
        <meta property="og:title" content="BMMB 554 / 2017">
        <meta property="og:url" content="https://nekrut.github.io/BMMB554/post/topic6/">
        <meta property="og:description" content="Introduction to data driven life sciences">
    
        <meta property="og:type" content="article" />
        <meta property="og:article:author" content="Anton Nekrutenko" />
        <meta property="og:article:published_time" content="2017-02-01T10:39:09-04:00" />
    
        <meta name="generator" content="Hugo 0.18.1" />
        <title>6. Introduction to Galaxy: Exons and Repeats &middot; BMMB 554 / 2017</title>
        <link rel="canonical" href="https://nekrut.github.io/BMMB554/" />
        <link rel="alternate" type="application/rss+xml" title="RSS" href="">
        <link rel="stylesheet" type="text/css" href="https://nekrut.github.io/BMMB554/css/main.css"/>
        <link href="//fonts.googleapis.com/css?family=Source+Sans+Pro:200|Montserrat:700" rel="stylesheet" type="text/css">
        <link href="//netdna.bootstrapcdn.com/font-awesome/4.0.3/css/font-awesome.css" rel="stylesheet">
        <script src="//code.jquery.com/jquery-1.10.2.min.js"></script>
    </head>

<body>
<!--[if lt IE 7]><p class="browsehappy">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> or <a href="http://www.google.com/chrome/â€Ž">install Google Chrome</a> to experience this site.</p><![endif]-->

    <header id="site-header">
        <div class="container">
            <a href="https://nekrut.github.io/BMMB554/" alt="BMMB 554 / 2017"><h1 class="blog-title heading">BMMB 554 / 2017</h1></a>
            <p class="blog-description">Introduction to data driven life sciences</p>
        </div>
    </header>
<main class="content" role="main">
	<div class="container">
		<article class="post">
	<header class="post-header">
        <h3 class="p-post-title">6. Introduction to Galaxy: Exons and Repeats</h3>
    </header>

    <section class="post-content">
        

<p><img src="/BMMB554/img/topic6_cover.png" alt="" /></p>

<h1 id="introduction-to-galaxy">Introduction to Galaxy</h1>

<p>In this lecture we will introduce you to bare basics of Galaxy:</p>

<ul>
<li>Getting data from external databases such as UCSC</li>
<li>Performing simple data manipulation</li>
<li>Understanding Galaxy&rsquo;s History system</li>
<li>Creating a running a workflow</li>
</ul>

<h2 id="what-are-we-trying-to-do">What are we trying to do?</h2>

<p>Suppose you get the following question:</p>

<blockquote>
<p><em>Mom (or Dad) &hellip; Which coding exon has the highest number of single nucleotide polymorphisms on chromosome 22?</em></p>
</blockquote>

<p>You think to yourself &ldquo;Wow! This is a simple question &hellip; I know exactly where the data is (at UCSC) but how do I actually compute this?&rdquo; The truth is, there is really no straightforward way of answering this question in a time frame comparable to the attention span of a 7-year-old. Well &hellip; actually there is and it is called Galaxy. So let&rsquo;s try it&hellip;</p>

<h2 id="0-organizing-your-windows-and-setting-up-galaxy-account">0. Organizing your windows and setting up Galaxy account</h2>

<h3 id="0-0-getting-your-display-sorted-out">0.0. Getting your display sorted out</h3>

<blockquote>
<p><strong>Note</strong>: Some screenshots shown here may appear slightly different from the ones you will see on your screen. Galaxy is quickly evolving and as a result some discrepancies are possible (and likely).</p>
</blockquote>

<p>To get the most of this tutorial open two browser windows. One you already have (it is this page). To open the other, <strong>right</strong> click <a href="http://usegalaxy.org">this link</a> and choose &ldquo;Open in a New Window&rdquo; (or something similar depending on your operating system and browser):</p>

<p><img src="http://galaxy.psu.edu/galaxy101/newWindow.png" alt="open in a new window" /></p>

<p>Then organize your windows as something like this (depending on the size of your monitor you may or may not be able to organize things this way, but you get the idea):</p>

<p><img src="https://galaxyproject.org/galaxy101/twoScreens.png" alt="Windows side by side" /></p>

<h3 id="0-1-setting-up-galaxy-account">0.1. Setting up Galaxy account</h3>

<p>Go to the <strong>User</strong> link at the top of Galaxy interface and choose <strong>Register</strong> (unless of course you already have an account):</p>

<p><img src="http://galaxy.psu.edu/galaxy101/register.png" alt="register" /></p>

<p>Then enter your information and you&rsquo;re in!</p>

<h2 id="1-getting-data-from-ucsc">1. Getting data from UCSC</h2>

<h3 id="1-0-getting-coding-exons">1.0. Getting coding exons</h3>

<p>First thing we will do is to obtain data from UCSC by clicking <strong>Get Data &#8594; UCSC Main</strong>:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/getDataUCSC.png" alt="get data from UCSC" /></p>

<p>You will see UCSC Table Browser interface appearing in your browser window:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/ucscGenes.png" alt="UCSC genes" /></p>

<p>Make sure that your settings are exactly the same as shown on the screen (in particular, <strong>position</strong> should be set to &ldquo;chr22&rdquo;, <strong>output format</strong> should be set to &ldquo;BED - browser extensible data&rdquo;, and &ldquo;Galaxy&rdquo; should be checked within the <strong>Send output to</strong> option). Click <strong>get output</strong> and you will see the next screen:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/ucscGenes2.png" alt="UCSC ganes 2" /></p>

<p>here make sure <strong>Create one BED record per:</strong> is set to &ldquo;Coding Exons&rdquo; and click <strong>Send Query to Galaxy</strong> button. After this you will see your first History Item in Galaxy&rsquo;s right pane. It will go through gray (preparing) and yellow (running) states to become green:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/firstHistoryItem.png" alt="First history item" /></p>

<h3 id="1-1-getting-snps">1.1. Getting SNPs</h3>

<p>Now is the time to obtain SNP data (SNPs are <a href="https://ghr.nlm.nih.gov/primer/genomicresearch/snp"><em>single nucleotide polymorphisms</em></a>). This is done almost exactly the same way. First thing we will do is to again click on <strong>Get Data &#8594; UCSC Main</strong>:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/getDataUCSC.png" alt="get data from UCSC" /></p>

<p>but now change <strong>group</strong> to &ldquo;Variation&rdquo;:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/variation.png" alt="Variation" /></p>

<p>so that the whole page looks like this:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/ucscSNPs.png" alt="UCSC SNPs" /></p>

<p>click get output and you should see this:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/ucscSNPs2.png" alt="UCSC SNPs 2" /></p>

<p>where you need to make sure that <strong>Whole Gene</strong> is selected (&ldquo;Whole Gene&rdquo; here really means &ldquo;Whole Feature&rdquo;) and click <strong>Send Query to Galaxy</strong> button. You will get your second item in the history:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/secondHistoryItem.png" alt="Second history item" /></p>

<p>Now we will rename the two history items to &ldquo;Exons&rdquo; and &ldquo;SNPs&rdquo; by clicking on the Pencil icon adjacent to each item. After changing the name scroll down and click <strong>Save</strong>.  Also we will rename history to &ldquo;my example&rdquo; (or whatever you want) by clicking on <strong>Unnamed history</strong> so everything looks like this:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/rename.png" alt="Rename" /></p>

<h3 id="video">Video</h3>


<div style="position: relative; padding-bottom: 56.25%; padding-top: 30px; height: 0; overflow: hidden;">
  <iframe src="//player.vimeo.com/video/185523444" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
 </div>


<h2 id="2-finding-exons-with-the-highest-number-of-snps">2. Finding Exons with the highest number of SNPs</h2>

<h3 id="2-0-joining-exons-with-snps">2.0. Joining exons with SNPs</h3>

<p>Let&rsquo;s remind ourselves that our objective was to find which exon contains the most SNPs. This first step in answering this question will be joining exons with SNPs (a fancy word for printing exons and SNPs that overlap side by side). This is done using <strong>Operate on Genomics Intervals &#8594; Join</strong> tool:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/join.png" alt="Join" /></p>

<p>make sure your <strong>Exons</strong> are first and <strong>SNPs</strong> are second and click <strong>Execute</strong>. You will get the third history item:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/thirdHistoryItem.png" alt="Third history item" /></p>

<h3 id="2-1-counting-the-number-of-snps-per-exon">2.1. Counting the number of SNPs per exon</h3>

<p>Let&rsquo;s look at the data obtained from the join operation above (you can do it by clicking the &ldquo;eye&rdquo; icon adjacent to the dataset). Below is a subsample of rows from the joined datasets (you may need to scroll sideways to see the entire length of the rows below):</p>

<hr />

<p><font color="red"></p>

<pre><code>chr22 15710867 15711034 uc062bej.1_cds_9_0_chr22_15710868_f  0 + chr22 15710880 15710881 rs568292779 0 -
chr22 15710867 15711034 uc062bej.1_cds_9_0_chr22_15710868_f  0 + chr22 15710947 15710948 rs544633418 0 -
chr22 15710867 15711034 uc062bej.1_cds_9_0_chr22_15710868_f  0 + chr22 15710906 15710907 rs548691624 0 -
chr22 15710867 15711034 uc062bej.1_cds_9_0_chr22_15710868_f  0 + chr22 15710920 15710921 rs530488686 0 -
chr22 15710867 15711034 uc062bej.1_cds_9_0_chr22_15710868_f  0 + chr22 15710932 15710933 rs563306354 0 -
chr22 15710867 15711034 uc062bej.1_cds_9_0_chr22_15710868_f  0 + chr22 15711019 15711020 rs559431407 0 -
chr22 15710867 15711034 uc062bej.1_cds_9_0_chr22_15710868_f  0 + chr22 15710949 15710950 rs532940301 0 -
</code></pre>

<p></font></p>

<pre><code>                                                      ....
</code></pre>

<pre><code>chr22 15719659 15719777 uc062bej.1_cds_10_0_chr22_15719660_f 0 + chr22 15719668 15719669 rs200891952 0 -
chr22 15719659 15719777 uc062bej.1_cds_10_0_chr22_15719660_f 0 + chr22 15719751 15719752 rs556077431 0 -
</code></pre>

<hr />

<p>Look at the rows highlighted in red. They all correspond to the same exon (id = <code>uc062bej.1_cds_9_0_chr22_15710868_f</code>) that overlaps seven distinct SNPs (ids: <code>rs568292779</code>, <code>rs544633418</code>, <code>rs548691624</code>, <code>rs530488686</code>, <code>rs563306354</code>, <code>rs559431407</code>, <code>rs532940301</code>). In other words this means that this exon contains seven SNPs. Since our ultimate goal is to count the number of SNPs per exon we can simply do this by counting how many times an exon&rsquo;s id appears in pur dataset. This can be easily done with the <strong>Join, Subtract, and Group &#8594; Group</strong> tool:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/group1.png" alt="Group" /></p>

<p>choose column 4 by selecting &ldquo;Column: 4&rdquo; in <strong>Group by</strong> column. Then click on <strong>Insert Operation</strong> and make sure the interface looks exactly as shown below:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/group2.png" alt="Group2" /></p>

<p>click <strong>Execute</strong>. Your history will look like this:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/fourthHistoryItem.png" alt="Fourth history item" /></p>

<p>if you look at the above image you will see that the result of grouping (dataset #4) contains two columns. This first contains the exon name while the second shows the number of times this name has been repeated in dataset #3.</p>

<h3 id="2-2-sorting-exons-by-snp-count">2.2. Sorting exons by SNP count</h3>

<p>To see which exon has the highest number of SNPs we can simply sort the dataset #4 on the second column in descending order. This is done with <strong>Filter and Sort &#8594; Sort</strong>:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/sort.png" alt="Sort" /></p>

<p>This will generate the fifth history item:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/fifthHistoryItem.png" alt="Fifth history item" /></p>

<p>and you can now see that the highest number of SNPs per exon is 63.</p>

<h3 id="2-3-selecting-top-five">2.3. Selecting top five</h3>

<p>Now let&rsquo;s select top five exons with the highest number of SNPs. For this we will use <strong>Text Manipulation &#8594; Select First</strong> tool:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/selectFirst.png" alt="Select first" /></p>

<p>Clicking <strong>Execute</strong> will produce the sixth history item that will contain just five lines:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/sixthHistoryItem.png" alt="Sixth history item" /></p>

<h3 id="2-4-recovering-exon-info-and-displaying-data-in-genome-browsers">2.4. Recovering exon info and displaying data in genome browsers</h3>

<p>Now we know that in this dataset the five top exons contain between 26 and 63 SNPs. But what else can we learn about these? To know more we need to get back the positional information (coordinates) of these exons. This information was lost at the grouping step and now all we have is just two columns. To get coordinates back we will match the names of exons in dataset #6 (column 1) against names of the exons in the original dataset #1 (column 4). This can be done with <strong>Join, Subtract and Group &#8594; Compare two Queries</strong> tool (note the settings of the tool in the middle pane):</p>

<p><img src="http://galaxy.psu.edu/galaxy101/compare.png" alt="Compare" /></p>

<p>this adds the seventh dataset to the history:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/seventhHistoryItem.png" alt="Seventh history item" /></p>

<p>The best way to learn about these exons is to look at their genomic surrounding. There is really no better way to do this than using genome browsers. Because this analysis was performed on &ldquo;standard&rdquo; human genome (<code>hg38</code> in this case), you have two choices - <a href="http://genomes.ucsc.edu">UCSC Genome Browser</a> and <a href="https://www.broadinstitute.org/igv/">IGV</a>:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/browsers.png" alt="" /></p>

<p>For example, clicking on <strong>display at UCSC main</strong> will show something like this:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/ucsc.png" alt="" /></p>

<h3 id="video-1">Video</h3>


<div style="position: relative; padding-bottom: 56.25%; padding-top: 30px; height: 0; overflow: hidden;">
  <iframe src="//player.vimeo.com/video/185538367" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
 </div>


<h2 id="3-understanding-histories">3. Understanding histories</h2>

<p>In Galaxy your analysis steps are represented as a list called <em>History</em>:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/seventhHistoryItem.png" alt="" /></p>

<p>Histories can be very large, you can have as many histories as you want, and all history behavior is controlled by the <img src="http://galaxyproject.org/galaxy101/fa-refresh.png" alt="refresh" />, <img src="http://galaxyproject.org/galaxy101/fa-cog.png" alt="cog" />, and <img src="http://galaxyproject.org/galaxy101/fa-columns.png" alt="multi_history" /> buttons on the top of the History pane:</p>

<p><img src="http://galaxyproject.org/galaxy101/historyDetail.png" alt="History detail" /></p>

<p>The <img src="http://galaxyproject.org/galaxy101/fa-refresh.png" alt="refresh" /> simply refreshes the history. The <img src="http://galaxyproject.org/galaxy101/fa-cog.png" alt="cog" /> button gives you access to myriad of history-specific options:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/historyOptions.png" alt="History options" /></p>

<p>Many of the options here are self-explanatory. If you create a new history, your current history does not disappear. If you would like to list all of your histories just choose <strong>Saved Histories</strong> and you will see a list of all your histories in the center pane:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/historyList.png" alt="History list" /></p>

<p>Yet there is a better way to look for all your histories. This is what the <img src="http://galaxyproject.org/galaxy101/fa-columns.png" alt="refresh" /> button is for. It allows you to browse and move datasets across histories:</p>

<p><img src="http://galaxyproject.org/galaxy101/multiHistoryView.png" alt="History list" /></p>

<p>Here, the current history is on the left (<strong>Galaxy 101 (2015)</strong>) and your (or mine in this case) other histories are displayed to the right of the current history. These can be ordered in a variety of ways by clicking the <strong>&hellip;</strong> button:</p>

<p><img src="http://galaxyproject.org/galaxy101/orderingHistories.png" alt="History list" /></p>

<p>You can also scroll sideways using trackpad gestures, move datasets across histories by simply clicking and dragging, and search for histories and individual datasets. This interface also allows you to switch to any existing history (i.e., making it current). Click <strong>Done</strong> once you&rsquo;re done.</p>

<p>A comprehensive overview of Galaxy&rsquo;s history functions is found <a href="https://wiki.galaxyproject.org/Histories">here</a>.</p>

<h2 id="4-creating-and-editing-a-workflow">4. Creating and editing a workflow</h2>

<h3 id="4-0-extracting-a-workflow">4.0. Extracting a workflow</h3>

<p>Lets take a look at the history again:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/historyCollapsed.png" alt="Collapsed history" /></p>

<p>You can see that this history contains all steps of our analysis. So by building this history we have actually created a complete record of our analysis with Galaxy preserving all parameter settings applied at every step. Wouldn&rsquo;t it be nice to just convert this history into a workflow that we&rsquo;ll be able to execute again and again? This can be done by clicking on the <img src="http://galaxyproject.org/galaxy101/fa-cog.png" alt="cog" /> button and selecting <strong>Extract Workflow</strong> option:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/extractWorkflow.png" alt="Extract workflow" /></p>

<p>The center pane will change as shown below and you will be able to choose which steps to include/exclude and how to name the newly created workflow. In this case I named it <code>galaxy101-2015</code>:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/createWorkflow.png" alt="Create workflow" /></p>

<p>once you click <strong>Create Workflow</strong> you will get the following message: &ldquo;Workflow &lsquo;galaxy101-2015&rsquo; created from current history. You can <strong>edit</strong> or <strong>run</strong> the workflow&rdquo;.</p>

<h3 id="4-1-opening-workflow-editor">4.1. Opening workflow editor</h3>

<p>Let&rsquo;s click <strong>edit</strong> (if you click something else and the message in the center page disappears, you can always access all your workflows including the one you just created using the <strong>Workflow</strong> link on top of Galaxy interface). This will open Galaxy&rsquo;s workflow editor (to get this view I clicked the arrow at the lower left corner of the screen, which collapsed the tool pane of the Galaxy interface). It will allow you to examine and change settings of this workflow as shown below. Note that the box corresponding to the <strong>Select First</strong> tool is selected (highlighted with the blueish border) and you can see parameters of this tool on the right pane. This is how you can view and change parameters of all tools involved in the workflow:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/wfEditor.png" alt="Workflow editor" /></p>

<h3 id="4-2-hiding-intermediate-steps">4.2. Hiding intermediate steps</h3>

<p>Among multiple things you can do with workflows I will just mention one. When workflow is executed one is usually interested in the final product and not in the intermediate steps. These steps can be hidden by mousing over a small asterisk in the lower right corner of every tool:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/hideStep.png" alt="Hide step" /></p>

<p>Yet there is a catch. In a newly created workflow all steps are hidden by default and the default behavior of Galaxy is that if all steps of a given workflow are hidden, then nothing gets hidden in the history. This may be counterintuitive, but this is done to decrease the amount of clicking if you do want to hide some steps. So in our case if we want to hide all intermediate steps with the exception of the last one we will click that asterisk in last step of the workflow:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/lastStep.png" alt="Last step" /></p>

<p>Once you do this the representation of the workflow in the bottom right corner of the editor will change with the last step becoming orange. This means that this is the only step, which will generate a dataset visible in the history:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/workflowOverview.png" alt="Workflow Overview" /></p>

<h3 id="4-3-renaming-inputs">4.3. Renaming inputs</h3>

<p>Right now both inputs to the workflow look exactly the same. This is a problem as will be very confusing which input should be <strong>Exons</strong> and which should be <strong>SNPs</strong>:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/namingInputs1.png" alt="Naming inputs 1" /></p>

<p>One the image above you will see that the top input dataset (the one with the blue border) connects to the <strong>Join tool</strong> first, so it must correspond to the exon data. If you click on this box (in the image above it is already clicked on because it is outlined with the blue border) you will be able to rename the dataset in the right pane:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/renameExons.png" alt="Rename Exons" /></p>

<p>Then click on the second input dataset and rename it &ldquo;Features&rdquo; (this would make this workflow a bit more generic, which will be useful later in this tutorial):</p>

<p><img src="http://galaxy.psu.edu/galaxy101/renameFeatures.png" alt="Rename features" /></p>

<h3 id="4-4-renaming-outputs">4.4. Renaming outputs</h3>

<p>Finally let&rsquo;s rename the workflow&rsquo;s output. For this:</p>

<ul>
<li>click on the last dataset (<strong>Compare two Queries</strong>)</li>
<li>scroll down the rightmost pane and click on <img src="http://galaxy.psu.edu/galaxy101/addAction.png" alt="add action" /></li>
<li>Type <code>Top Exons</code> in the <strong>Rename dataset</strong> text box:</li>
</ul>

<p><img src="http://galaxy.psu.edu/galaxy101/topExons.png" alt="Top exons" /></p>

<h3 id="4-5-setting-parameters-at-runtime">4.5. Setting parameters &ldquo;at runtime&rdquo;</h3>

<p>What we are trying to do here is do design a generic workflow. This means that time to time you will need to change parameters within this workflow. For instance, in this tutorial we were selecting 5 exons containing the highest number of SNPs. But what if you need to select 10? Thus it makes sense to leave these types of parameters adjustable. To do this:</p>

<p>First, select a tool in which you want to set parameters at runtime (<code>Select first</code> in this case):</p>

<p><img src="http://galaxy.psu.edu/galaxy101/runtimeTool.png" alt="runtime Tool Selection" /></p>

<p>Next, select parameter you would like to set at runtime. To do this just hover over the <img src="http://galaxy.psu.edu/galaxy101/paramSymbol.png" alt="paramSymbol" /> icon so it looks like this:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/runtimeParam.png" alt="runtime Param Selection" /></p>

<p>and click! Your parameter will now be set at runtime.</p>

<h3 id="4-6-save-it-is-important">4.6. Save! It is important&hellip;</h3>

<p>Now let&rsquo;s save the changes we&rsquo;ve made by clicking <img src="http://galaxyproject.org/galaxy101/fa-cog.png" alt="cog" /> and selecting Save:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/wfSave.png" alt="wfSave" /></p>

<h2 id="5-run-workflow-on-whole-genome-data">5. Run workflow on whole genome data</h2>

<p>Now that we have a workflow, let&rsquo;s do something grand like, for example, finding exons with the highest number of repetitive elements across the entire human genome.</p>

<h3 id="5-0-create-a-new-history">5.0. Create a new history</h3>

<p>First go back into analysis view by clicking <strong>Analyze Data</strong> on top of the Galaxy&rsquo;s interface. Now let&rsquo;s create a new history by clicking <img src="http://galaxyproject.org/galaxy101/fa-cog.png" alt="cog" /> and selecting <strong>Create New</strong>:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/createNewHistory.png" alt="Create new history" /></p>

<h3 id="5-1-get-exons">5.1. Get Exons</h3>

<p>Now let&rsquo;s get coding exons for the entire genome by going to <strong>Get Data &#8594; UCSC Main</strong> and setting up parameters as shown below. Note that this time <code>region</code> radio button is set to <strong>genome</strong>:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/getAllGenes.png" alt="Get all genes" /></p>

<p>Click <strong>get output</strong> and you will get the next page (if it looks different from the image below, go back and make sure <code>output format</code> is set to <strong>BED - browser extensible format</strong>):</p>

<p><img src="http://galaxy.psu.edu/galaxy101/ucscGenes2.png" alt="Get exons" /></p>

<p>Choose <strong>Coding exons</strong> and click <strong>Send query to Galaxy</strong>.</p>

<h2 id="5-2-get-repeats">5.2. Get Repeats</h2>

<p>Go again to <strong>Get Data &#8594; UCSC Main</strong> and make sure the following settings are selected (in particular <code>group</code> = <strong>Repeats</strong> and <code>track</code> = <strong>RepeatMasker</strong>):</p>

<p><img src="http://galaxy.psu.edu/galaxy101/allRepeats.png" alt="All repeats" /></p>

<p>Click <strong>get output</strong> and you will get the next page (if it looks different from the image below, go back and make sure <code>output format</code> is set to <strong>BED - browser extensible format</strong>):</p>

<p><img src="http://galaxy.psu.edu/galaxy101/allRepeats2.png" alt="All repeats 2" /></p>

<p>Select <strong>Whole gene</strong> and click <strong>Send Query to Galaxy</strong>.</p>

<h3 id="5-3-start-the-workflow">5.3. Start the Workflow</h3>

<p>At this point you will have two items in your history - one with exons and one with repeats. These datasets are large (especially repeats) and it will take some time for them to become green. Luckily you do not have to wait as Galaxy will automatically start jobs once uploads have ended. So nothing stops us from starting the workflow we have created. First, click on the <strong>Workflow link</strong> at the top of Galaxy interface, mouse over <strong>galaxy101-2015</strong>, click, and select <strong>Run</strong>. Center pane will change to allow you launching the workflow. Select appropriate datasets for <code>Repeats</code> and <code>Exon</code> inputs as shown below. Now scroll to <strong>Step 6</strong> and will see that we can set up <code>Select first</code> parameter at <em>Runtime</em> (meaning Now!). So lets put <code>20</code> in there (or anything else you want) and scroll further down to click <img src="http://galaxy.psu.edu/galaxy101/runWorkflowButton.png" alt="Run workflow" /> to see this:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/launchWorkflow.png" alt="Launch workflow" /></p>

<p>Once workflow has started you will initially be able to see all its steps. Note that you are joining all exons with all repeats, so naturally this will take some time:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/launchedWorkflow.png" alt="Launched workflow" /></p>

<h3 id="5-4-get-coffee">5.4. Get coffee</h3>

<p>As we mentioned above this will take some time, so go get coffee. At last you will see this:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/final.png" alt="Final view" /></p>

<h2 id="6-we-did-not-fake-this">6. We did not fake this:</h2>

<p>The two histories and the workflow described in this page are accessible directly from this page below:</p>

<ul>
<li>History <a href="https://usegalaxy.org/u/aun1/h/galaxy-101-2015"><strong>Galaxy 101 (2015)</strong></a></li>
<li>History <a href="https://usegalaxy.org/u/aun1/h/exons-vs-repeats-2015-1"><strong>Exons vs. Repeats</strong></a></li>
<li>Workflow <a href="https://usegalaxy.org/u/aun1/w/galaxy101-2015-1"><strong>Galaxy 101-2015</strong></a></li>
</ul>

<p>From there you can import histories and workflows to make them your own. For example, to import <strong>Galaxy 101 (2015)</strong> history simply click <a href="https://usegalaxy.org/u/aun1/h/galaxy-101-2015">this link</a> and select <code>Import history</code> link:</p>

<p><img src="http://galaxy.psu.edu/galaxy101/importHistory.png" alt="Final view" /></p>

<h1 id="next">Next</h1>

<p>We will start using Galaxy to process and map NGS data.</p>

    </section>

    <hr>

    <footer class="post-footer">
        <section class="f-1">
            
            
            <p class="f-post-time"><time datetime="2017-02-01T10:39:09-04:00">February 1, 2017</time></p>
        </section>
                        
        <section class="f-2">
            <section class="share">
                <span>Share:
                <a class="icon-twitter" href="http://twitter.com/share?text=6.%20Introduction%20to%20Galaxy%3a%20Exons%20and%20Repeats&url=https%3a%2f%2fnekrut.github.io%2fBMMB554%2fpost%2ftopic6%2f"
                    onclick="window.open(this.href, 'twitter-share', 'width=550,height=235');return false;">
                    <i class="fa fa-twitter"></i>
                </a>
                <a class="icon-facebook" href="https://www.facebook.com/sharer/sharer.php?u=https%3a%2f%2fnekrut.github.io%2fBMMB554%2fpost%2ftopic6%2f"
                    onclick="window.open(this.href, 'facebook-share','width=580,height=296');return false;">
                    <i class="fa fa-facebook"></i>
                </a>
                <a class="icon-google-plus" href="https://plus.google.com/share?url=https%3a%2f%2fnekrut.github.io%2fBMMB554%2fpost%2ftopic6%2f"
                   onclick="window.open(this.href, 'google-plus-share', 'width=490,height=530');return false;">
                    <i class="fa fa-google-plus"></i>
                </a>
                </span>
            </section>

            
        </section>
                        
    </footer>
</article>
	</div>
</main>
    <footer id="site-footer">
    <hr>
        <div class="container">
            
            <section><a href="https://nekrut.github.io/BMMB554/">Anton Nekrutenko</a> 2017 | MIT License</section>
        </div>
    </footer>

    <script type="text/javascript" src="https://nekrut.github.io/BMMB554/js/fittext.js"></script>
    <script type="text/javascript">
      $(".heading").fitText();
    </script>
    <script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/x-mathjax-config">
  MathJax.Hub.Queue(function() {
    
    
    
    var all = MathJax.Hub.getAllJax(), i;
    for(i = 0; i < all.length; i += 1) {
        all[i].SourceElement().parentNode.className += ' has-jax';
    }
});
</script>


</body>
</html>