<!doctype html>

<html lang="en-us">

<head>
  <title>BMMB 554 | 2018</title>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="description" content="The HTML5 Herald" />
  <meta name="author" content="Anton Nekrutenko" />
  <meta name="generator" content="Hugo 0.32.3" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/normalize/7.0.0/normalize.min.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" />
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto+Slab|Ruda" />
  <link rel="stylesheet" type="text/css" href="https://nekrut.github.io/BMMB554/css/styles.css" />
</head>

<body>
  <div id="container">
    <header>
      <h1>
                <a href="https://nekrut.github.io/BMMB554/">BMMB 554 | 2018</a>
            </h1>

      <ul id="social-media">
        
        <li><a href="https://twitter.com/galaxyproject"><i class="fa fa-twitter fa-lg" aria-hidden="true"></i></a></li>
          
        <li><a href="https://github.com/nekrut"><i class="fa fa-github fa-lg" aria-hidden="true"></i></a></li>
           
      </ul>
      
      <p><em>Introduction to data driven life sciences</em></p>
      
    </header>

    
<nav>
    <ul>
        
        <li>
            <a class="" href="https://nekrut.github.io/BMMB554/about/">
                <i class="fa-li fa  fa-lg"></i><span>Syllabus</span>
            </a>
        </li>
        
        <li>
            <a class="" href="https://nekrut.github.io/BMMB554/posts/">
                <i class="fa-li fa  fa-lg"></i><span>Lectures</span>
            </a>
        </li>
        
    </ul>
</nav>

    <main>




<article>

    <h1>DNA Sequencing: Illumina and PacBio</h1>

    
        <aside>
    <ul>
        <li>
            <time class="post-date" datetime="2018-01-22T00:00:00Z">Jan 22, 2018</time>
        </li>
        
        

        
        <li>
            <em>
                
                    
                    <a href="https://nekrut.github.io/BMMB554/tags/dna-sequencing/">#DNA sequencing</a>
                
                    , 
                    <a href="https://nekrut.github.io/BMMB554/tags/illumina/">#Illumina</a>
                
            </em>
        </li>
        

        <li>4 min read</li>
    </ul>
</aside>
    

    

<h1 id="a-few-classical-papers-to-start">A few classical papers to start</h1>

<p>In a series of now classical papers Philip Green and co-workers has developed a quantitative framework for the analysis of data generated by automated DNA sequencers.</p>

<table>
<thead>
<tr>
<th></th>
</tr>
</thead>

<tbody>
<tr>
<td><a href="http://genome.cshlp.org/content/8/3/175.full"><img src="https://nekrut.github.io/BMMB554/img/ewing_green1.png" alt="" /></a></td>
</tr>

<tr>
<td><a href="http://genome.cshlp.org/content/8/3/186.full"><img src="https://nekrut.github.io/BMMB554/img/ewing_green2.png" alt="" /></a></td>
</tr>

<tr>
<td><small><strong>Two papers</strong> were published back to back in <em>Genome Research</em></td>
</tr>
</tbody>
</table>

<p>In particular they developed a standard metric for describing the reliability of base calls:</p>

<blockquote>
<p>An important technical aspect of our work is the use of log-transformed error probabilities rather than untransformed ones, which facilitates working with error rates in the range of most importance (very close to 0). Specifically, we define the quality value q assigned to a base-call to be<br><br>
$q = -10\times log(p)$<br><br>
where p is the estimated error probability for that base-call. Thus a base-call having a probability of <sup>1</sup>&frasl;<sub>1000</sub> of being incorrect is assigned a quality value of 30. Note that high quality values correspond to low error probabilities, and conversely.</p>
</blockquote>

<p>We will be using the concept of &ldquo;<em>quality score</em>&rdquo; or &ldquo;<em>phred-scaled quality score</em>&rdquo; repeatedly in this course.</p>

<h1 id="illumina">Illumina</h1>

<p><img src="https://nekrut.github.io/BMMB554/img/illumina_pseudocolor.png" alt="" /></p>

<p>Illumina (originally called &ldquo;Solexa&rdquo;) uses glass flowcells with oligonucleotides permanently attached to internal surface. These oligonucleotides are complementary to sequencing adapters added to DNA fragments being sequenced during library preparation. The DNA fragments that are &ldquo;stuck&rdquo; on the flowcell due to complementary interaction between adapters are amplified via &ldquo;bridge amplification&rdquo; to form clusters. Sequencing is performed using reversible terminator chemistry with nucleotides modified to carry dyes specific to each base. As a result all nucleotides can be added at once and are distinguished by color. Currently, it is possible to sequence up to 300 bases from each end of the fragment being sequenced. Illumina has the highest throughput (and lowest cost per base) of all existing technologies at this moment. The NovaSeq 6000 machine can produce <a href="https://www.illumina.com/systems/sequencing-platforms.html">6000 billion nucleotides in 44 hours</a>. In this course we will most often work with Illumina data.</p>

<h3 id="slides">Slides</h3>

<script async class='speakerdeck-embed' data-id='942908c8c24546d58cf8b61b3598feb3' data-ratio='1.33333333333333' src='//speakerdeck.com/assets/embed.js'></script>

<h3 id="video">Video</h3>


<div style="position: relative; padding-bottom: 56.25%; padding-top: 30px; height: 0; overflow: hidden;">
  <iframe src="//player.vimeo.com/video/121178846" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
 </div>


<h3 id="reading">Reading</h3>

<ul>
<li>2008 | <a href="http://www.nature.com/nature/journal/v456/n7218/pdf/nature07517.pdf">Human genome sequencing on Illumina - Bentley et al.</a></li>
<li>2010 | <a href="http://nar.oxfordjournals.org/content/39/13/e90.full-text-lowres.pdf">Data quality 1 - Nakamura et al.</a></li>
<li>2011 | <a href="http://genomebiology.com/content/pdf/gb-2011-12-11-r112.pdf">Data quality 2 - Minoche et al.</a></li>
<li>2011 | <a href="http://www.biomedcentral.com/content/pdf/1471-2164-12-382.pdf">Illumina pitfalls - Kircher et al.</a></li>
</ul>

<h3 id="10x-a-way-to-extend-the-utility-of-short-illumina-reads">10X: A way to extend the utility of short Illumina reads</h3>

<p>A company called <a href="http://www.10xgenomics.com/technology/">10X Genomics</a> has developed a technology which labels reads derived from continuous fragments of genomic DNA. In this technology a gel bead covered with a large number of adapter molecules is placed within a droplet containing PCR reagents and one or several long molecules of genomic DNA to be sequenced.</p>

<table>
<thead>
<tr>
<th></th>
</tr>
</thead>

<tbody>
<tr>
<td><img src="https://nekrut.github.io/BMMB554/img/10x.png" alt="" /></td>
</tr>

<tr>
<td><small><strong>10X bead</strong> containing the standard Illumina P5 adapter combined with barcode, sequencing primer annealing site (R1) and random primer (N-mer) Slide from <a href="http://www.slideshare.net/GenomeInABottle/aug2015-analysis-team-04-10x-genomics">Slideshare</a>.</small></td>
</tr>
</tbody>
</table>

<p>The droplets are generated by combining beads, PCR reagents, and genomic DNA in a microfluidic device:</p>

<table>
<thead>
<tr>
<th></th>
</tr>
</thead>

<tbody>
<tr>
<td><a href="http://www.nature.com/nbt/journal/v34/n3/fig_tab/nbt.3432_F1.html"><img src="https://nekrut.github.io/BMMB554/img/10x-overview.jpg" alt="" /></a></td>
</tr>

<tr>
<td><small><strong>10X workflow</strong> (<strong>a</strong>) Gel beads loaded with primers and barcoded oligonucleotides are mixed with DNA and enzyme mixture then oil-surfactant solution at a microfluidic &lsquo;double-cross&rsquo; junction. Gel beadâ€“containing droplets flow to a reservoir where gel beads are dissolved, initiating whole-genome primer extension. The products are pooled from each droplet. The final library preparation requires shearing the libraries and incorporation of Illumina adapters. (<strong>b</strong>) Top, linked reads of the ALK gene from the NA12878 WGS sample. Lines represent linked reads; dots represent reads; color indicates barcode. Middle, exon boundaries of the ALK gene. Bottom, linked reads of the ALK gene from the NA12878 exome data. Reads from neighboring exons are linked by common barcodes. Only a small fraction of linked reads is presented here. Reproduced from <a href="http://www.nature.com/nbt/journal/v34/n3/full/nbt.3432.html">Zheng:2015</a> (click the image to go to the original paper).</small></td>
</tr>
</tbody>
</table>

<p>In essence, 10X allows to uniquely map reads derived from long genomic fragments. This information is essential for bridging together genome and transcriptome assemblies as we will see in later in this course.</p>


<div style="position: relative; padding-bottom: 56.25%; padding-top: 30px; height: 0; overflow: hidden;">
  <iframe src="//player.vimeo.com/video/120429438" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
 </div>


<h2 id="pacbio-single-molecule-sequencing">PacBio Single Molecule Sequencing</h2>

<p>PacBio is a fundamentally different approach to DNA sequencing as it allows reading single molecules. Thus it is an example is so called <em>Single Molecule Sequencing</em> or <em>SMS</em>. PacBio uses highly processive DNA polymerase placed at the bottom of each well on a microtiter plate. The plate is fused to the glass slide illuminated by a laser. When polymerase is loaded with template it attracts fluorescently labeled nucleotides to the bottom of the well where they emit light with a wavelength characteristic of each nucleotide. As a result a &ldquo;movie&rdquo; is generated for each well recording the sequence and duration of incorporation events. One of the key advantage of PacBio technology is its ability to produce long reads with ones at 10,000 bases being common.</p>

<h3 id="slides-1">Slides</h3>

<script async class='speakerdeck-embed' data-id='ccb3d34f1a214f66ac7a2d233caedaf5' data-ratio='1.33333333333333' src='//speakerdeck.com/assets/embed.js'></script>

<h3 id="video-1">Video</h3>


<div style="position: relative; padding-bottom: 56.25%; padding-top: 30px; height: 0; overflow: hidden;">
  <iframe src="//player.vimeo.com/video/121267426" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
 </div>


<p>Underlying slides are <a href="https://speakerdeck.com/nekrut/ngs-technologies-pacific-biosceinces">here</a></p>

<h3 id="reading-1">Reading</h3>

<ul>
<li>2003 | <a href="http://www.sciencemag.org/content/299/5607/682.full.pdf">Single Molecule Analaysis at High Concentration - Levene et al.</a></li>
<li>2008 | <a href="http://www.pnas.org/content/105/4/1176.full">ZMW nanostructures - Korlach et al.</a></li>
<li>2009 | <a href="http://www.sciencemag.org/content/323/5910/133.full">Real Time Sequencing with PacBio - Eid et al.</a></li>
<li>2010 | <a href="http://www.nature.com/nmeth/journal/v7/n6/pdf/nmeth.1459.pdf">Modification detection with PacBio - Flusberg et al.</a></li>
<li>2012 | <a href="http://www.nature.com/nbt/journal/v30/n7/pdf/nbt.2280.pdf">Error correction of PacBio reads - Koren et al.</a></li>
<li>2014 | <a href="http://www.pnas.org/cgi/doi/10.1073/pnas.1400447111">Transcriptome with PacBio - Taligner et al.</a></li>
<li>2015 | <a href="http://dx.doi.org/10.1038/nature13907">Resolving complex regions in Human genomes with PacBio - Chaisson et al.</a></li>
</ul>


</article>


<section class="post-nav">
    <ul>
        
        <li>
            <a href="https://nekrut.github.io/BMMB554/posts/topic2_1/"><i class="fa fa-chevron-circle-left"></i> DNA Sequecning: From Sanger to 454</a>
        </li>
        
        
    </ul>
</section>
    





</main>
    <footer>
        <h6>Anton Nekrutenko | 2018 | CC-BY </h6>
    </footer>
</div>
<script src="https://nekrut.github.io/BMMB554/js/scripts.js"></script>
<script type="text/javascript" async
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
  });
  MathJax.Hub.Queue(function() {
    
    
    
    var all = MathJax.Hub.getAllJax(), i;
    for(i = 0; i < all.length; i += 1) {
        all[i].SourceElement().parentNode.className += ' has-jax';
    }
  });

  MathJax.Hub.Config({
  
  TeX: { equationNumbers: { autoNumber: "AMS" } }
  });
</script>

</body>
</html>


