+++
categories = []
date = "2018-01-22"
draft = true
menu = ""
tags = ['DNA sequencing', 'Illumina']
title = "3. Illumina: Process and Data"
+++

![](/img/illumina_pseudocolor.png)

## 2: Illumina sequencing

Illumina (originally called "Solexa") uses glass flowcells with oligonucleotides permanently attached to internal surface. These oligonucleotides are complementary to sequencing adapters added to DNA fragments being sequenced during library preparation. The DNA fragements that are "stuck" on the flowcell due to complementary interaction between adapters are amplified via "bridge amplification" to form clusters. Sequencing is performed using reversible terminator chemistry with nucleotides modified to carry dyes specific to each nucleotide. As a result all nucleotides can be added at once and are distinguished by colors. Currently, it is possible to sequence up to 300 bases from each end of the fragment being sequenced. Illumina has the highest throughput (and lowest cost per base) of all existing technologies at this moment. The HiSeq 2500 machine can produce [600 billion nucleotides in 5 days](http://www.illumina.com/systems/hiseq_2500_1500/performance_specifications.html). In this course we will most often work with Illumina data.

### Slides

{{< speakerdeck 942908c8c24546d58cf8b61b3598feb3 >}}

### Video

{{< vimeo 121178846 >}}

Underlying slides are [here](https://speakerdeck.com/nekrut/ngs-technologies-illumina#)

### Reading

* 2008 | [Human genome sequencing on Illumina - Bentley et al.](http://www.nature.com/nature/journal/v456/n7218/pdf/nature07517.pdf)
* 2010 | [Data quality 1 - Nakamura et al.](http://nar.oxfordjournals.org/content/39/13/e90.full-text-lowres.pdf)
* 2011 | [Data quality 2 - Minoche et al.](http://genomebiology.com/content/pdf/gb-2011-12-11-r112.pdf)
* 2011 | [Illumina pitfalls - Kircher et al.](http://www.biomedcentral.com/content/pdf/1471-2164-12-382.pdf)

### 10X: A way to extend the utility of short Illumina reads

A company called [10X Genomics](http://www.10xgenomics.com/technology/) has developed a technology which labels reads derived from continuous fragments of genomic DNA. In this technology a gel bead covered with a large number of adapter molecules is placed within a droplet containing PCR reagents and one or several long molecules of genomic DNA to be sequenced. 

> ![](/img/10x.png)
>
> **10X bead** containing the standard Illumina P5 adapter combined with barcode, sequencing primer annealing site (R1) and random primer (N-mer) Slide from [Slideshare](http://www.slideshare.net/GenomeInABottle/aug2015-analysis-team-04-10x-genomics).

The droplets are generated by combining beads, PCR reagents, and genomic DNA in a microfluidic device:

> [![](/img/10x-overview.jpg)](http://www.nature.com/nbt/journal/v34/n3/fig_tab/nbt.3432_F1.html)
>
> **10X workflow** (**a**) Gel beads loaded with primers and barcoded oligonucleotides are mixed with DNA and enzyme mixture then oil-surfactant solution at a microfluidic 'double-cross' junction. Gel beadâ€“containing droplets flow to a reservoir where gel beads are dissolved, initiating whole-genome primer extension. The products are pooled from each droplet. The final library preparation requires shearing the libraries and incorporation of Illumina adapters. (**b**) Top, linked reads of the ALK gene from the NA12878 WGS sample. Lines represent linked reads; dots represent reads; color indicates barcode. Middle, exon boundaries of the ALK gene. Bottom, linked reads of the ALK gene from the NA12878 exome data. Reads from neighboring exons are linked by common barcodes. Only a small fraction of linked reads is presented here. Reproduced from [Zheng:2015](http://www.nature.com/nbt/journal/v34/n3/full/nbt.3432.html) (click the image to go to the original paper).

In essence, 10X allows to uniquely map reads derived from long genomic fragments. This information is essential for bridging together genome and transcriptome assemblies as we will see in later in this course. 

{{< vimeo 120429438 >}}


## 3: PacBio Single Molecule Sequencing

PacBio is a fundamentally different approach to DNA sequencing as it allows reading single molecules. Thus it is an example is so called _Single Molecule Sequencing_ or _SMS_. PacBio uses highly processive DNA polymerase placed at the bottom of each well on a microtiter plate. The plate is fused to the glass slide illuminated by a laser. When polymerase is loaded with template it attracts fluorescently labeled nucleotides to the bottom of the well where they emit light with a wavelength characteristic of each nucleotide. As a result a "movie" is generated for each well recording the sequence and duration of incorporation events. One of the key advantage of PacBio technology is its ability to produce long reads with ones at 10,000 bases being common. 

### Slides

{{< speakerdeck ccb3d34f1a214f66ac7a2d233caedaf5 >}}

### Video

{{< vimeo 121267426 >}}

Underlying slides are [here](https://speakerdeck.com/nekrut/ngs-technologies-pacific-biosceinces)

### Reading

* 2003 | [Single Molecule Analaysis at High Concentration - Levene et al.](http://www.sciencemag.org/content/299/5607/682.full.pdf)
* 2008 | [ZMW nanostructures - Korlach et al.](http://www.pnas.org/content/105/4/1176.full)
* 2009 | [Real Time Sequencing with PacBio - Eid et al.](http://www.sciencemag.org/content/323/5910/133.full)
* 2010 | [Modification detection with PacBio - Flusberg et al.](http://www.nature.com/nmeth/journal/v7/n6/pdf/nmeth.1459.pdf)
* 2012 | [Error correction of PacBio reads - Koren et al.](http://www.nature.com/nbt/journal/v30/n7/pdf/nbt.2280.pdf)
* 2014 | [Transcriptome with PacBio - Taligner et al.](http://www.pnas.org/cgi/doi/10.1073/pnas.1400447111)
* 2015 | [Resolving complex regions in Human genomes with PacBio - Chaisson et al.](http://dx.doi.org/10.1038/nature13907)

## 4: Oxford Nanopore

Oxford nanopore is another dramatically different technology that threads single DNA molecules through biologically-derived (transmembrane proteins) pore in a membrane impermeable to ions. It uses polymerase to control the speed of translocation of the DNA molecule through membrane. In that sense it is not _Sequencing by synthesis_ we have seen in the other technologies discussed here. This technology generates longest reads possible today: in many instances a single read can be hundreds of thousands if nucleotides in length. It still however suffers from high error rate and relatively low throughput (compared to Illumina). On the upside Oxford Nanopore sequencing machines are only slightly bigger than a thumb-drive and cost very little. 

### Slides

{{< speakerdeck 3895a3069bc64ad5bc74c972a0353911 >}}

A recent overview of latest developments at Oxford Nanopore can be found [here](https://github.com/lmmx/talk-transcripts/blob/master/Nanopore/NoThanksIveAlreadyGotOne.md) as well as in the following video

{{< youtube nizGyutn6v4 >}}

### Reading

* 2016 | [The promises and challenges of solid-state sequencing](http://nature.com/nnano/journal/v11/n2/full/nnano.2016.9.html)
* 2015 | [Improved data analysis for the minion nanopore sequencer](http://nature.com/nmeth/journal/v12/n4/full/nmeth.3290.html)
* 2015 | [MinION Analysis and Reference Consortium](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4722697/)
* 2015 | [A complete bacterial genome assembled *de novo* using only nanopore sequencing data](http://www.nature.com/nmeth/journal/v12/n8/full/nmeth.3444.html)
* 2012 | [Reading DNA at single-nucleotide resolution with a mutant MspA nanopore and phi29 DNA polymerase](http://nature.com/nbt/journal/v30/n4/full/nbt.2171.html)
* Simpson Lab [blog](http://simpsonlab.github.io/2015/04/08/eventalign/)
* Poretools analysis [suite](http://poretools.readthedocs.org/)
