<!DOCTYPE html>
<!--[if lt IE 7]><html class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
<!--[if IE 7]><html class="no-js lt-ie9 lt-ie8"> <![endif]-->
<!--[if IE 8]><html class="no-js lt-ie9"> <![endif]-->
<!--[if gt IE 8]><!--><html class="no-js" prefix="og: http://ogp.me/ns#" xmlns:og="http://ogp.me/ns#"><!--<![endif]-->

    <head>
                <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="description" content="Introduction to data driven life sciences">
        <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=1.0" />
        <meta name="mobile-web-app-capable" content="yes">
        <meta name="apple-mobile-web-app-capable" content="yes">
        <meta name="apple-mobile-web-app-status-bar-style" content="black">
        <meta name="author" content="Anton Nekrutenko">
  
	
        <meta property="og:site_name" content="BMMB 554 / 2017">
        <meta property="og:title" content="BMMB 554 / 2017">
        <meta property="og:url" content="https://nekrut.github.io/BMMB554/">
        <meta property="og:description" content="Introduction to data driven life sciences">
    
        <meta property="og:type" content="website" />
    
        <meta name="generator" content="Hugo 0.18.1" />
        <title>BMMB 554 / 2017 &middot; BMMB 554 / 2017</title>
        <link rel="canonical" href="https://nekrut.github.io/BMMB554/" />
        <link rel="alternate" type="application/rss+xml" title="RSS" href="https://nekrut.github.io/BMMB554/index.xml">
        <link rel="stylesheet" type="text/css" href="https://nekrut.github.io/BMMB554/css/main.css"/>
        <link href="//fonts.googleapis.com/css?family=Source+Sans+Pro:200|Montserrat:700" rel="stylesheet" type="text/css">
        <link href="//netdna.bootstrapcdn.com/font-awesome/4.0.3/css/font-awesome.css" rel="stylesheet">
        <script src="//code.jquery.com/jquery-1.10.2.min.js"></script>
    </head>

<body>
<!--[if lt IE 7]><p class="browsehappy">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> or <a href="http://www.google.com/chrome/â€Ž">install Google Chrome</a> to experience this site.</p><![endif]-->

    <header id="site-header">
        <div class="container">
            <a href="https://nekrut.github.io/BMMB554/" alt="BMMB 554 / 2017"><h1 class="blog-title heading">BMMB 554 / 2017</h1></a>
            <p class="blog-description">Introduction to data driven life sciences</p>
        </div>
    </header>
<main class="content" role="main">
	<div class="container">
	
		<article class="li post">
    <header class="post-header">
        <h3 class="post-title"><a href="https://nekrut.github.io/BMMB554/post/topic10/">11. Finding really low frequency variants</a></h3>
        <p class="post-time"><time datetime="2017-02-20T17:19:07-05:00">February 20, 2017</time></p>
    </header>
    
    <section class="post-excerpt">
        <p>This page explains how to perform discovery of low frequency variants from duplex sequencing data. As an example we use the ABL1 dataset published by Schmitt and colleagues (SRA accession SRR1799908).
Background Calling low frequency variants from next generation sequencing (NGS) data is challenging due to significant amount of noise characteristic of these technologies. Duplex sequencing (DS) was designed to address this problem by increasing sequencing accuracy by over four orders of magnitude.</p>
    </section>

    <footer class="post-footer">
        <span>
            
            
        </span>
    </footer>
</article>
	
		<article class="li post">
    <header class="post-header">
        <h3 class="post-title"><a href="https://nekrut.github.io/BMMB554/post/topic9a/">10. Why calling variants?</a></h3>
        <p class="post-time"><time datetime="2017-02-20T10:11:10-04:00">February 20, 2017</time></p>
    </header>
    
    <section class="post-excerpt">
        <p>Tibet is high Many residents of the Tibetian Plateau live above 4,000 meters where oxygen concentration is approximately 40% lower than at sea level:
 Tibetan Plateau and surrounding areas (from Wikipedia).
 Tibetian experience a number of adaptations to high altitudes manifesting in the following phenotypic differences:
 lower hemoglobin concentration Wu:2005 higher arterial oxygen saturation Niermeyer:1995 more efficient pulmonary gas exchange Zhuang:1996  What are the genetic causes of these adaptations?</p>
    </section>

    <footer class="post-footer">
        <span>
            
            
        </span>
    </footer>
</article>
	
		<article class="li post">
    <header class="post-header">
        <h3 class="post-title"><a href="https://nekrut.github.io/BMMB554/post/topic9/">9. Non-diploid variant calling</a></h3>
        <p class="post-time"><time datetime="2017-02-13T10:11:10-04:00">February 13, 2017</time></p>
    </header>
    
    <section class="post-excerpt">
        <p>The majority of life on Earth is non-diploid and represented by prokaryotes, viruses and their derivatives such as our own mitochondria or plant&rsquo;s chloroplasts. In non-diploid systems allele frequencies can range anywhere between 0 and 100% and there could be multiple (not just two) alleles per locus. The main challenge associated with non-diploid variant calling is the difficulty in distinguishing between sequencing noise (abundant in all NGS platforms) and true low frequency variants.</p>
    </section>

    <footer class="post-footer">
        <span>
            
            
        </span>
    </footer>
</article>
	
		<article class="li post">
    <header class="post-header">
        <h3 class="post-title"><a href="https://nekrut.github.io/BMMB554/post/topic8/">8. Diploid variant calling</a></h3>
        <p class="post-time"><time datetime="2017-02-12T11:42:45-04:00">February 12, 2017</time></p>
    </header>
    
    <section class="post-excerpt">
        <p>Today we hear a lot about personalized medicine. Yet the personalization is defined by the genetic make up of the individual. Today we will discuss how this information can be uncovered from the genomic sequencing data. The figure above shows distribution of rare and common variants in 1,092 human genomes described by the 1000 Genome Consortium.
Calling variants Variant calling is a complex field that was significantly propelled by advances in DNA sequencing and efforts of large scientific consortia such as the 1000 Genomes.</p>
    </section>

    <footer class="post-footer">
        <span>
            
            
        </span>
    </footer>
</article>
	
		<article class="li post">
    <header class="post-header">
        <h3 class="post-title"><a href="https://nekrut.github.io/BMMB554/post/topic7/">7. NGS data: Practicalities</a></h3>
        <p class="post-time"><time datetime="2017-02-08T11:26:49-04:00">February 8, 2017</time></p>
    </header>
    
    <section class="post-excerpt">
        <p>In this section we will look at practical aspects of manipulation of next-generation sequencing data. We will start with Fastq format produced by most sequencing machines and will finish with SAM/BAM format representing mapped reads. The cover image above shows a screen dump of a SAM dataset.
Getting NGS data in You can data in Galaxy using one of five ways:
From your computer This works well for small files because web browser do not like lengthy file transfers:</p>
    </section>

    <footer class="post-footer">
        <span>
            
            
        </span>
    </footer>
</article>
	
		<article class="li post">
    <header class="post-header">
        <h3 class="post-title"><a href="https://nekrut.github.io/BMMB554/post/topic6/">6. Introduction to Galaxy: Exons and Repeats</a></h3>
        <p class="post-time"><time datetime="2017-02-01T10:39:09-04:00">February 1, 2017</time></p>
    </header>
    
    <section class="post-excerpt">
        <p>Introduction to Galaxy In this lecture we will introduce you to bare basics of Galaxy:
 Getting data from external databases such as UCSC Performing simple data manipulation Understanding Galaxy&rsquo;s History system Creating a running a workflow  What are we trying to do? Suppose you get the following question:
 Mom (or Dad) &hellip; Which coding exon has the highest number of single nucleotide polymorphisms on chromosome 22?</p>
    </section>

    <footer class="post-footer">
        <span>
            
            
        </span>
    </footer>
</article>
	
		<article class="li post">
    <header class="post-header">
        <h3 class="post-title"><a href="https://nekrut.github.io/BMMB554/post/topic5/">5. Aligning many sequences quickly</a></h3>
        <p class="post-time"><time datetime="2017-01-30T10:13:02-04:00">January 30, 2017</time></p>
    </header>
    
    <section class="post-excerpt">
        <p>Speeding things up The topics we discussed in the past lecture explain fundamental concepts behind analysis of biological sequences. Today, we will be talking about algorithms that allow aligning billions of sequencing reads against reference genomes. Similarly to the previous lecture I have borrowed heavily from the course taught by Ben Langmead at Johns Hopkins. The cover image is from Wikpedia article on Burrows-Wheeler transform.
The challenge of really large datasets In the previous lecture we have seen how dynamic programming helps aligning sequences.</p>
    </section>

    <footer class="post-footer">
        <span>
            
            
            	&nbsp;&nbsp;<i class="fa fa-tag"></i>
            	mapping, alignment, burrows-wheeler transform
            
        </span>
    </footer>
</article>
	
		<article class="li post">
    <header class="post-header">
        <h3 class="post-title"><a href="https://nekrut.github.io/BMMB554/post/topic4/">4. Aligning sequences with dynamic programming</a></h3>
        <p class="post-time"><time datetime="2017-01-25T11:12:04-04:00">January 25, 2017</time></p>
    </header>
    
    <section class="post-excerpt">
        <p>Sequence alignment In the previous lecture we have seen the principle behind dynamic programming. This approach is extremely useful for comparing biological sequences, which is coincidentally one of the main points of this course. This lecture explain how this is done. In writing this text I heavily relied on wonderful course taught by Ben Langmead at Johns Hopkins. The cover image shows pairwise alignments for human, mouse, and dog KIF3 locus from Dubchak et al.</p>
    </section>

    <footer class="post-footer">
        <span>
            
            
            	&nbsp;&nbsp;<i class="fa fa-tag"></i>
            	global alignment, local alignment, dynamic programming
            
        </span>
    </footer>
</article>
	
		<article class="li post">
    <header class="post-header">
        <h3 class="post-title"><a href="https://nekrut.github.io/BMMB554/post/topic3/">3. Sequence alignment: Introductory concepts</a></h3>
        <p class="post-time"><time datetime="2017-01-25T10:43:18-04:00">January 25, 2017</time></p>
    </header>
    
    <section class="post-excerpt">
        <p>Introduction In the previous lecture we have seen several ways in which DNA sequence data can be accumulated (the reason for having Manhattan in the figure above will be apparent a bit later). Because sequencing machines (especially the ones made by Illumina) generate billions of sequences (called reads) from every run, the real challenge is what one does with all this data once sequencing is done. So before we get into details of technology and its application we need to introduce some basic algorithmic concepts related to sequence analysis.</p>
    </section>

    <footer class="post-footer">
        <span>
            
            
            	&nbsp;&nbsp;<i class="fa fa-tag"></i>
            	dynamic programming
            
        </span>
    </footer>
</article>
	
		<article class="li post">
    <header class="post-header">
        <h3 class="post-title"><a href="https://nekrut.github.io/BMMB554/post/topic2/">2. DNA sequencing</a></h3>
        <p class="post-time"><time datetime="2017-01-17T11:17:20-04:00">January 17, 2017</time></p>
    </header>
    
    <section class="post-excerpt">
        <p>The 60s and the 70s The first complete nucleic acids being sequenced were RNAs (tRNAs in particular; see pioneering work of Robert Holley and colleagues). The work on finding approaches to sequencing DNA molecules began in late 60s and early 70s. One of the earliest contributions has been made by Ray Wu from Cornell, who used E. coli DNA polymerase to incorporate radioactively labelled nucleotides into protruding ends of bacteriphage lambda.</p>
    </section>

    <footer class="post-footer">
        <span>
            
            
            	&nbsp;&nbsp;<i class="fa fa-tag"></i>
            	NGS, sequencing, DNA
            
        </span>
    </footer>
</article>
	
	
<nav role="pagination" class="pagination">
  
	<span class="post-list-pagination-item post-list-pagination-item-current">Page 1 of 2</span>

  <a href="/bmmb554/page/2/" class="older-posts">
    &nbsp;Older&nbsp;<i class="fa fa-angle-double-right"></i>
  </a>

</nav>

	</div>
</main>
    <footer id="site-footer">
    <hr>
        <div class="container">
            
            <section><a href="https://nekrut.github.io/BMMB554/">Anton Nekrutenko</a> 2017 | MIT License</section>
        </div>
    </footer>

    <script type="text/javascript" src="https://nekrut.github.io/BMMB554/js/fittext.js"></script>
    <script type="text/javascript">
      $(".heading").fitText();
    </script>
    <script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/x-mathjax-config">
  MathJax.Hub.Queue(function() {
    
    
    
    var all = MathJax.Hub.getAllJax(), i;
    for(i = 0; i < all.length; i += 1) {
        all[i].SourceElement().parentNode.className += ' has-jax';
    }
});
</script>

</body>
</html>